<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />

<meta name="date" content="2018-07-13" />

<title>Important Transcript-Level Estimates</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">weight_restoration</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Important Transcript-Level Estimates</h1>
<h4 class="author"><em>Lauren Blake</em></h4>
<h4 class="date"><em>2018-07-13</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-08-12</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> 241c630 </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/figure/
    Ignored:    data/.DS_Store
    Ignored:    data/aux_info/
    Ignored:    data/hg_38/
    Ignored:    data/libParams/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  _workflowr.yml
    Untracked:  analysis/Collection_dates.Rmd
    Untracked:  analysis/Converting_IDs.Rmd
    Untracked:  analysis/Global_var_hg38.Rmd
    Untracked:  analysis/Global_variation.Rmd
    Untracked:  analysis/Preliminary_clinical_covariate.Rmd
    Untracked:  analysis/Technical_confounders.Rmd
    Untracked:  analysis/VennDiagram2018-07-24_06-55-46.log
    Untracked:  analysis/VennDiagram2018-07-24_06-56-13.log
    Untracked:  analysis/VennDiagram2018-07-24_06-56-50.log
    Untracked:  analysis/VennDiagram2018-07-24_06-58-41.log
    Untracked:  analysis/VennDiagram2018-07-24_07-00-07.log
    Untracked:  analysis/VennDiagram2018-07-24_07-00-42.log
    Untracked:  analysis/VennDiagram2018-07-24_07-01-08.log
    Untracked:  analysis/View_DE_genes.Rmd
    Untracked:  analysis/analysis.Rmd
    Untracked:  analysis/correlations_bet_covariates.Rmd
    Untracked:  analysis/correlations_over_time.Rmd
    Untracked:  analysis/genocode_annotation_info.Rmd
    Untracked:  analysis/import_transcript_level_estimates.Rmd
    Untracked:  analysis/tximport.Rmd
    Untracked:  analysis/unnormalized_data.Rmd
    Untracked:  analysis/variables_by_time.Rmd
    Untracked:  data/405-BAN22201T1-MH0163673_AATCCAGC_S9_L001_R1_001/
    Untracked:  data/405-BAN22201T2-MH0163857_CGTCTAAC_S3_L004_R1_001/
    Untracked:  data/405-BAN22201T3-MH0167352_CTAAGACC_S31_L004_R1_001/
    Untracked:  data/405-BAN22202T1-MH0163700_CTTCCTTC_S19_L002_R1_001/
    Untracked:  data/405-BAN22202T2-MH0164426_GCACACAA_S4_L004_R1_001/
    Untracked:  data/405-BAN22202T3-MH0167384_ACTCCTAC_S18_L002_R1_001/
    Untracked:  data/405-BAN22203T1-MH0163897_TCCATTGC_S32_L004_R1_001/
    Untracked:  data/405-BAN22203T2-MH0164104_CTAAGACC_S21_L003_R1_001/
    Untracked:  data/405-BAN22204T1-MH0164041_CTTCCTTC_S19_L005_R1_001/
    Untracked:  data/405-BAN22204T2-MH0165543_CTCCTAGT_S8_L001_R1_001/
    Untracked:  data/405-BAN22204T3-MH0168921_AACTTGCC_S57_L006_R1_001/
    Untracked:  data/405-BAN22205T1-MH0164218_CGCTACAT_S7_L004_R1_001/
    Untracked:  data/405-BAN22205T2-MH0165296_AACTTGCC_S17_L002_R1_001/
    Untracked:  data/405-BAN22206T1-MH0164595_CGCTACAT_S7_L001_R1_001/
    Untracked:  data/405-BAN22206T2-MH0165241_GTCGAGAA_S6_L001_R1_001/
    Untracked:  data/405-BAN22206T3-MH0167362_CTCCTAGT_S68_L008_R1_001/
    Untracked:  data/405-BAN22207T1-MH0164655_ACAACAGC_S10_L001_R1_001/
    Untracked:  data/405-BAN22207T2-MH0165541_CCTTAGGT_S22_L003_R1_001/
    Untracked:  data/405-BAN22207T3-MH0168576_CGCTACAT_S75_L008_R1_001/
    Untracked:  data/405-BAN22208T1-MH0165002_CTCGAACA_S30_L003_R1_001/
    Untracked:  data/405-BAN22208T2-MH0165266_ATCGTCTC_S29_L003_R1_001/
    Untracked:  data/405-BAN22208T3-MH0168126_ACAACAGC_S48_L005_R1_001/
    Untracked:  data/405-BAN22209T1-MH0165265_AACTCGGA_S2_L004_R1_001/
    Untracked:  data/405-BAN22209T2-MH0165750_TCTAGGAG_S28_L003_R1_001/
    Untracked:  data/405-BAN22209T3-MH0168523_CGTCCATT_S16_L005_R1_001/
    Untracked:  data/405-BAN22210T1-MH0165547_AACCGAAC_S27_L003_R1_001/
    Untracked:  data/405-BAN22210T2-MH0166544_AACGCCTT_S26_L003_R1_001/
    Untracked:  data/405-BAN22210T3-MH0168587_CTTCCTTC_S56_L006_R1_001/
    Untracked:  data/405-BAN22212T1-MH0165601_GCACACAA_S4_L001_R1_001/
    Untracked:  data/405-BAN22212T2-MH0165753_CGTCCATT_S16_L002_R1_001/
    Untracked:  data/405-BAN22215T1-MH0165895_GACTACGA_S15_L005_R1_001/
    Untracked:  data/405-BAN22215T2-MH0166334_AATCCAGC_S9_L004_R1_001/
    Untracked:  data/405-BAN22215T3-MH0168486_CCTTAGGT_S32_L004_R1_001/
    Untracked:  data/405-BAN22216T1-MH0165949_CAAGCCAA_S31_L004_R1_001/
    Untracked:  data/405-BAN22216T2-MH0167360_GACTACGA_S15_L002_R1_001/
    Untracked:  data/405-BAN22216T3-MH0169379_CGTCTAAC_S47_L005_R1_001/
    Untracked:  data/405-BAN22218T1-MH0166539_GTACACCT_S14_L002_R1_001/
    Untracked:  data/405-BAN22218T2-MH0167895_CGACCTAA_S13_L002_R1_001/
    Untracked:  data/405-BAN22218T3-MH0169739_AATCCAGC_S14_L002_R1_001/
    Untracked:  data/405-BAN22219T1-MH0167403_CGTCTAAC_S3_L001_R1_001/
    Untracked:  data/405-BAN22219T2-MH0168012_ACTCCTAC_S18_L005_R1_001/
    Untracked:  data/405-BAN22219T3-MH0170053_CGTCCATT_S55_L006_R1_001/
    Untracked:  data/405-BAN22220T1-MH0167615_CCTATACC_S25_L003_R1_001/
    Untracked:  data/405-BAN22220T2-MH0168005_CCAACACT_S24_L003_R1_001/
    Untracked:  data/405-BAN22220T3-MH0169956_AACTCGGA_S2_L001_R1_001/
    Untracked:  data/405-BAN22221T1-MH0168112_ACCATCCT_S11_L005_R1_001/
    Untracked:  data/405-BAN22221T2-MH0168529_TACATCGG_S33_L004_R1_001/
    Untracked:  data/405-BAN22221T3-MH0171129_GTCGAGAA_S46_L005_R1_001/
    Untracked:  data/405-BAN22222T1-MH0168671_CTAAGACC_S21_L006_R1_001/
    Untracked:  data/405-BAN22222T2-MH0169515_TCTTCGAC_S74_L008_R1_001/
    Untracked:  data/405-BAN22224T1-MH0169098_CTTCCTTC_S73_L008_R1_001/
    Untracked:  data/405-BAN22224T2-MH0169571_GACTACGA_S21_L003_R1_001/
    Untracked:  data/405-BAN22224T3-MH0172869_AACCGAAC_S34_L004_R1_001/
    Untracked:  data/405-BAN22226T1-MH0169516_AACTTGCC_S22_L003_R1_001/
    Untracked:  data/405-BAN22226T2-MH0170691_ACGAGAAC_S23_L003_R1_001/
    Untracked:  data/405-BAN22226T3-MH0173165_GCACACAA_S24_L003_R1_001/
    Untracked:  data/405-BAN22228T1-MH0169654_CGTCCATT_S29_L003_R1_001/
    Untracked:  data/405-BAN22228T2-MH0171900_ATGACAGG_S28_L003_R1_001/
    Untracked:  data/405-BAN22228T3-MH0174229_ACCATCCT_S27_L003_R1_001/
    Untracked:  data/405-BAN22229T1-MH0170475_CTCCTAGT_S26_L003_R1_001/
    Untracked:  data/405-BAN22229T2-MH0171083_GTCGAGAA_S13_L002_R1_001/
    Untracked:  data/405-BAN22232T1-MH0171809_AACTCGGA_S76_L008_R1_001/
    Untracked:  data/405-BAN22232T2-MH0171901_GACTACGA_S15_L008_R1_001/
    Untracked:  data/405-BAN22232T3-MH0174220_ATCGTCTC_S71_L008_R1_001/
    Untracked:  data/405-BAN22233T1-MH0171922_GTACACCT_S4_L001_R1_001/
    Untracked:  data/405-BAN22233T2-MH0172715_GTACACCT_S30_L003_R1_001/
    Untracked:  data/405-BAN22233T3-MH0175373_ACGGACTT_S35_L004_R1_001/
    Untracked:  data/405-BAN22234T1-MH0172376_ACAACAGC_S10_L007_R1_001/
    Untracked:  data/405-BAN22234T2-MH0174414_CGCTACAT_S12_L002_R1_001/
    Untracked:  data/405-BAN22234T3-MH0177050_AACTCGGA_S49_L005_R1_001/
    Untracked:  data/405-BAN22235T1-MH0172813_TCCATTGC_S11_L002_R1_001/
    Untracked:  data/405-BAN22235T2-MH0173158_CGTCTAAC_S70_L008_R1_001/
    Untracked:  data/405-BAN22235T3-MH0175276_AACGCCTT_S36_L004_R1_001/
    Untracked:  data/405-BAN22236T1-MH0173201_CGACCTAA_S37_L004_R1_001/
    Untracked:  data/405-BAN22236T2-MH0174188_ACAACAGC_S15_L002_R1_001/
    Untracked:  data/405-BAN22239T1-MH0174265_ACTCCTAC_S25_L003_R1_001/
    Untracked:  data/405-BAN22239T2-MH0174778_TACATCGG_S12_L002_R1_001/
    Untracked:  data/405-BAN22239T3-MH0178097_CCTATACC_S38_L004_R1_001/
    Untracked:  data/405-BAN22240T1-MH0174969_CGCTACAT_S7_L007_R1_001/
    Untracked:  data/405-BAN22240T2-MH0175802_AACTTGCC_S17_L008_R1_001/
    Untracked:  data/405-BAN22242T1-MH0175396_ATGACAGG_S5_L008_R1_001/
    Untracked:  data/405-BAN22242T2-MH0177006_ACTCCTAC_S18_L008_R1_001/
    Untracked:  data/405-BAN22242T3-MH0181773_CAAGCCAA_S44_L005_R1_001/
    Untracked:  data/405-BAN22243T1-MH0175481_CCTATACC_S20_L002_R1_001/
    Untracked:  data/405-BAN22243T2-MH0176559_ACGGACTT_S3_L001_R1_001/
    Untracked:  data/405-BAN22243T3-MH0179740_TCTAGGAG_S39_L004_R1_001/
    Untracked:  data/405-BAN22245T1-MH0177498_ATGACAGG_S5_L004_R1_001/
    Untracked:  data/405-BAN22245T2-MH0179482_CAAGCCAA_S31_L007_R1_001/
    Untracked:  data/405-BAN22247T1-MH0178383_TCTTCGAC_S1_L001_R1_001/
    Untracked:  data/405-BAN22247T2-MH0179693_ACCATCCT_S11_L002_R1_001/
    Untracked:  data/405-BAN22248T1-MH0179252_CCAACACT_S24_L006_R1_001/
    Untracked:  data/405-BAN22248T2-MH0179936_ACGGACTT_S23_L003_R1_001/
    Untracked:  data/405-BAN22248T3-MH0184420_CCAACACT_S69_L008_R1_001/
    Untracked:  data/405-BAN22249T1-MH0179995_TCTTCGAC_S1_L005_R1_001/
    Untracked:  data/405-BAN22249T2-MH0180995_CTCGAACA_S2_L001_R1_001/
    Untracked:  data/405-BAN22250T1-MH0180438_GTCGAGAA_S6_L004_R1_001/
    Untracked:  data/405-BAN22250T2-MH0181264_GCACACAA_S4_L008_R1_001/
    Untracked:  data/405-BAN22251T1-MH0181000_AACTCGGA_S2_L007_R1_001/
    Untracked:  data/405-BAN22251T2-MH0181244_CTCGAACA_S30_L006_R1_001/
    Untracked:  data/405-BAN22252T1-MH0181193_AACCGAAC_S19_L002_R1_001/
    Untracked:  data/405-BAN22252T2-MH0181540_ATGACAGG_S5_L001_R1_001/
    Untracked:  data/405-BAN22253T1-MH0181369_TACATCGG_S1_L001_R1_001/
    Untracked:  data/405-BAN22253T2-MH0183318_CTCCTAGT_S8_L005_R1_001/
    Untracked:  data/405-BAN22254T1-MH0183287_CCAACACT_S10_L001_R1_001/
    Untracked:  data/405-BAN22254T2-MH0183347_CCTTAGGT_S22_L006_R1_001/
    Untracked:  data/405-BAN22254T3-MH0186085_AATCCAGC_S72_L008_R1_001/
    Untracked:  data/405-BAN22255T1-MH0186017_CCTTAGGT_S18_L002_R1_001/
    Untracked:  data/405-BAN22255T2-MH0186145_GTCGAGAA_S6_L007_R1_001/
    Untracked:  data/405-BAN22256T1-MH0186190_ACGAGAAC_S20_L005_R1_001/
    Untracked:  data/405-BAN22256T2-MH0186757_CGTCCATT_S16_L008_R1_001/
    Untracked:  data/405-BAN22256T3-MH0192924_TCCATTGC_S43_L005_R1_001/
    Untracked:  data/405-BAN22257T1-MH0186793_CTCCTAGT_S8_L008_R1_001/
    Untracked:  data/405-BAN22257T2-MH0187462_TCCATTGC_S32_L007_R1_001/
    Untracked:  data/405-BAN22258T1-MH0187064_AATCCAGC_S9_L007_R1_001/
    Untracked:  data/405-BAN22258T2-MH0187605_CTAAGACC_S6_L001_R1_001/
    Untracked:  data/405-BAN22258T3-MH0198483_ACCATCCT_S54_L006_R1_001/
    Untracked:  data/405-BAN22260T1-MH0187236_GTACACCT_S14_L005_R1_001/
    Untracked:  data/405-BAN22260T2-MH0188240_CTTCCTTC_S19_L008_R1_001/
    Untracked:  data/405-BAN22260T3-MH0198647_ATGACAGG_S42_L005_R1_001/
    Untracked:  data/405-BAN22261T1-MH0188182_ACGGACTT_S23_L006_R1_001/
    Untracked:  data/405-BAN22261T2-MH0191258_CGACCTAA_S7_L001_R1_001/
    Untracked:  data/405-BAN22262T1-MH0188248_TCTTCGAC_S1_L008_R1_001/
    Untracked:  data/405-BAN22262T2-MH0191318_CCTATACC_S25_L007_R1_001/
    Untracked:  data/405-BAN22266T1-MH0192891_ACAACAGC_S10_L004_R1_001/
    Untracked:  data/405-BAN22266T2-MH0197478_ACGAGAAC_S20_L002_R1_001/
    Untracked:  data/405-BAN22266T3-MH0201141_GACTACGA_S53_L006_R1_001/
    Untracked:  data/405-BAN22267T1-MH0197522_AACTTGCC_S17_L005_R1_001/
    Untracked:  data/405-BAN22267T2-MH0197979_ATCGTCTC_S29_L006_R1_001/
    Untracked:  data/405-BAN22268T1-MH0197785_TCTAGGAG_S28_L006_R1_001/
    Untracked:  data/405-BAN22268T2-MH0197998_CGACCTAA_S13_L006_R1_001/
    Untracked:  data/405-BAN22268T3-MH0203576_ACTCCTAC_S52_L006_R1_001/
    Untracked:  data/405-BAN22269T1-MH0197823_CGTCTAAC_S3_L007_R1_001/
    Untracked:  data/405-BAN22269T2-MH0197945_ATCGTCTC_S8_L001_R1_001/
    Untracked:  data/405-BAN22270T1-MH0199425_CAAGCCAA_S17_L002_R1_001/
    Untracked:  data/405-BAN22270T2-MH0199506_ACCATCCT_S11_L008_R1_001/
    Untracked:  data/405-BAN22271T1-MH0200966_AACGCCTT_S26_L007_R1_001/
    Untracked:  data/405-BAN22271T2-MH0201765_AACCGAAC_S27_L006_R1_001/
    Untracked:  data/405-BAN22271T3-MH0217142_CTCGAACA_S40_L004_R1_001/
    Untracked:  data/405-BAN22272T1-MH0201784_TACATCGG_S12_L006_R1_001/
    Untracked:  data/405-BAN22272T2-MH0202184_AACGCCTT_S16_L002_R1_001/
    Untracked:  data/405-BAN22272T3-MH0217144_GCACACAA_S41_L005_R1_001/
    Untracked:  data/405-BAN22274T1-MH0201904_TCTAGGAG_S9_L001_R1_001/
    Untracked:  data/405-BAN22274T2-MH0202323_ACGAGAAC_S5_L001_R1_001/
    Untracked:  data/405-BAN22274T3-MH0217163_TCTTCGAC_S45_L005_R1_001/
    Untracked:  data/August_13_vsd_values_hg38_gc.rds
    Untracked:  data/BAK_BSP_001-BAN22202_RRED397T1_GTACACCT_S51_L006_R1_001/
    Untracked:  data/BAK_BSP_002-BAN22202_RRED397T2_ACGAGAAC_S50_L006_R1_001/
    Untracked:  data/BAK_BSP_003-BAN22220_RRED398T1_CGACCTAA_S58_L006_R1_001/
    Untracked:  data/BAK_BSP_004-BAN22220_RRED398T2_TACATCGG_S59_L007_R1_001/
    Untracked:  data/BAK_BSP_005-BAN22209_RRED399T1_ATCGTCTC_S60_L007_R1_001/
    Untracked:  data/BAK_BSP_006-BAN22209_RRED399T2_CCAACACT_S67_L007_R1_001/
    Untracked:  data/BAK_BSP_007-BAN22218_RRED424T1_TCTAGGAG_S66_L007_R1_001/
    Untracked:  data/BAK_BSP_008-BAN22218_RRED424T2_CTCGAACA_S65_L007_R1_001/
    Untracked:  data/BAK_BSP_009-BAN22226_RRED453T1_ACGGACTT_S64_L007_R1_001/
    Untracked:  data/BAK_BSP_010-BAN22226_RRED453T2_CTAAGACC_S63_L007_R1_001/
    Untracked:  data/BAK_BSP_011-BAN22228_RRED467T1_AACCGAAC_S62_L007_R1_001/
    Untracked:  data/BAK_BSP_012-BAN22228_RRED467T2_CCTTAGGT_S61_L007_R1_001/
    Untracked:  data/BAN2 Dates_T1_T2.xlsx
    Untracked:  data/BAN_DATES.csv
    Untracked:  data/BAN_DATES.xlsx
    Untracked:  data/BAN_DATES_txt.csv
    Untracked:  data/Blood_dates.txt
    Untracked:  data/Labels.csv
    Untracked:  data/Labels.xlsx
    Untracked:  data/RIN.xlsx
    Untracked:  data/RIN_over_time.csv
    Untracked:  data/RIN_over_time.xlsx
    Untracked:  data/T0_consolid.csv
    Untracked:  data/T0_consolid.xlsx
    Untracked:  data/age_t1.txt
    Untracked:  data/cmd_info.json
    Untracked:  data/counts_fc1.txt
    Untracked:  data/counts_fc1.txt.summary
    Untracked:  data/counts_hg38.RData
    Untracked:  data/counts_hg38_gc.RData
    Untracked:  data/files_list.txt
    Untracked:  data/final_covariates.csv
    Untracked:  data/gene_names_58387.txt
    Untracked:  data/logs/
    Untracked:  data/notimecovariates.csv
    Untracked:  data/only_individuals_biomarkers_weight_restoration_study.xlsx
    Untracked:  data/rest1t2_BI_hg37.rds
    Untracked:  data/rest1t2_BI_hg38.rds
    Untracked:  data/rest1t2_hg37.rds
    Untracked:  data/rest1t2_psych_meds_BMI_hg37.rds
    Untracked:  data/rest1t2_psych_meds_hg37.rds
    Untracked:  data/rest2t3_BI_hg37.rds
    Untracked:  data/rest2t3_BI_hg38.rds
    Untracked:  data/rest2t3_hg37.rds
    Untracked:  data/rest2t3_psych_meds_BMI_hg37.rds
    Untracked:  data/rest2t3_psych_meds_hg37.rds
    Untracked:  data/salmon_gene_matrix_bak_reorder_time.txt
    Untracked:  data/technical_sample_info.csv
    Untracked:  data/tx_to_gene.txt
    Untracked:  data/tx_to_gene_37.txt
    Untracked:  data/~$Labels.xlsx
    Untracked:  data/~$T0_consolid.xlsx
    Untracked:  docs/VennDiagram2018-07-24_06-55-46.log
    Untracked:  docs/VennDiagram2018-07-24_06-56-13.log
    Untracked:  docs/VennDiagram2018-07-24_06-56-50.log
    Untracked:  docs/VennDiagram2018-07-24_06-58-41.log
    Untracked:  docs/VennDiagram2018-07-24_07-00-07.log
    Untracked:  docs/VennDiagram2018-07-24_07-00-42.log
    Untracked:  docs/VennDiagram2018-07-24_07-01-08.log
    Untracked:  docs/figure/

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Deleted:    analysis/chunks.R
    Modified:   analysis/index.Rmd
    Modified:   analysis/license.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<hr />
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Import transcript-level estimates with the tximportData package.</p>
<pre class="r"><code>#source(&quot;https://bioconductor.org/biocLite.R&quot;)
#biocLite(&quot;EnsDb.Hsapiens.v75&quot;)

library(tximport)
library(readr)
library(DESeq2)</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, cbind, colMeans,
    colnames, colSums, do.call, duplicated, eval, evalq, Filter,
    Find, get, grep, grepl, intersect, is.unsorted, lapply,
    lengths, Map, mapply, match, mget, order, paste, pmax,
    pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce,
    rowMeans, rownames, rowSums, sapply, setdiff, sort, table,
    tapply, union, unique, unsplit, which, which.max, which.min</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: SummarizedExperiment</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>Loading required package: DelayedArray</code></pre>
<pre><code>Loading required package: matrixStats</code></pre>
<pre><code>
Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:Biobase&#39;:

    anyMissing, rowMedians</code></pre>
<pre><code>
Attaching package: &#39;DelayedArray&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    apply</code></pre>
<pre class="r"><code>library(cowplot)</code></pre>
<pre><code>Warning: package &#39;cowplot&#39; was built under R version 3.4.4</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 3.4.4</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<pre class="r"><code>library(&quot;topGO&quot;)</code></pre>
<pre><code>Loading required package: graph</code></pre>
<pre><code>Loading required package: GO.db</code></pre>
<pre><code>Loading required package: AnnotationDbi</code></pre>
<pre><code></code></pre>
<pre><code>Loading required package: SparseM</code></pre>
<pre><code>
Attaching package: &#39;SparseM&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    backsolve</code></pre>
<pre><code>
groupGOTerms:   GOBPTerm, GOMFTerm, GOCCTerm environments built.</code></pre>
<pre><code>
Attaching package: &#39;topGO&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    members</code></pre>
<pre class="r"><code>#library(&quot;biomaRt&quot;)
library(&quot;clusterProfiler&quot;)</code></pre>
<pre><code>Loading required package: DOSE</code></pre>
<pre><code>DOSE v3.4.0  For help: https://guangchuangyu.github.io/DOSE

If you use DOSE in published research, please cite:
Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609</code></pre>
<pre><code>clusterProfiler v3.6.0  For help: https://guangchuangyu.github.io/clusterProfiler

If you use clusterProfiler in published research, please cite:
Guangchuang Yu., Li-Gen Wang, Yanyan Han, Qing-Yu He. clusterProfiler: an R package for comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012, 16(5):284-287.</code></pre>
<pre class="r"><code>library(&quot;org.Hs.eg.db&quot;)</code></pre>
<pre><code></code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ──────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ tibble  1.4.2     ✔ dplyr   0.7.6
✔ tidyr   0.8.1     ✔ stringr 1.3.1
✔ purrr   0.2.5     ✔ forcats 0.3.0</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 3.4.4</code></pre>
<pre><code>Warning: package &#39;purrr&#39; was built under R version 3.4.4</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 3.4.4</code></pre>
<pre><code>Warning: package &#39;stringr&#39; was built under R version 3.4.4</code></pre>
<pre><code>── Conflicts ─────────────────────── tidyverse_conflicts() ──
✖ stringr::boundary() masks graph::boundary()
✖ dplyr::collapse()   masks IRanges::collapse()
✖ dplyr::combine()    masks Biobase::combine(), BiocGenerics::combine()
✖ dplyr::count()      masks matrixStats::count()
✖ dplyr::desc()       masks IRanges::desc()
✖ tidyr::expand()     masks S4Vectors::expand()
✖ dplyr::filter()     masks stats::filter()
✖ dplyr::first()      masks S4Vectors::first()
✖ cowplot::ggsave()   masks ggplot2::ggsave()
✖ dplyr::lag()        masks stats::lag()
✖ ggplot2::Position() masks BiocGenerics::Position(), base::Position()
✖ purrr::reduce()     masks GenomicRanges::reduce(), IRanges::reduce()
✖ dplyr::rename()     masks S4Vectors::rename()
✖ dplyr::select()     masks AnnotationDbi::select()
✖ purrr::simplify()   masks clusterProfiler::simplify()
✖ dplyr::slice()      masks IRanges::slice()</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>Warning: package &#39;data.table&#39; was built under R version 3.4.4</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre><code>The following object is masked from &#39;package:SummarizedExperiment&#39;:

    shift</code></pre>
<pre><code>The following object is masked from &#39;package:GenomicRanges&#39;:

    shift</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    shift</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, second</code></pre>
<pre class="r"><code>library(plyr)</code></pre>
<pre><code>-------------------------------------------------------------------------</code></pre>
<pre><code>You have loaded plyr after dplyr - this is likely to cause problems.
If you need functions from both plyr and dplyr, please load plyr first, then dplyr:
library(plyr); library(dplyr)</code></pre>
<pre><code>-------------------------------------------------------------------------</code></pre>
<pre><code>
Attaching package: &#39;plyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    arrange, count, desc, failwith, id, mutate, rename, summarise,
    summarize</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    compact</code></pre>
<pre><code>The following object is masked from &#39;package:graph&#39;:

    join</code></pre>
<pre><code>The following object is masked from &#39;package:matrixStats&#39;:

    count</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    desc</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    rename</code></pre>
<pre class="r"><code>library(&quot;dplyr&quot;)

#create a tx2gene.txt table
library(EnsDb.Hsapiens.v75)</code></pre>
<pre><code>Loading required package: ensembldb</code></pre>
<pre><code>Loading required package: GenomicFeatures</code></pre>
<pre><code>
Attaching package: &#39;GenomicFeatures&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:topGO&#39;:

    genes</code></pre>
<pre><code>Loading required package: AnnotationFilter</code></pre>
<pre><code>
Attaching package: &#39;ensembldb&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    filter</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
<pre class="r"><code>edb &lt;- EnsDb.Hsapiens.v75

Tx.ensemble &lt;- transcripts(edb,
          columns = c(&quot;tx_id&quot;, &quot;gene_id&quot;, &quot;gene_name&quot;),
          return.type = &quot;DataFrame&quot;)
nrow(Tx.ensemble)</code></pre>
<pre><code>[1] 215647</code></pre>
<pre class="r"><code>tx2gene&lt;- Tx.ensemble[,c(1:3)]
head(tx2gene)</code></pre>
<pre><code>DataFrame with 6 rows and 3 columns
            tx_id         gene_id   gene_name
      &lt;character&gt;     &lt;character&gt; &lt;character&gt;
1 ENST00000000233 ENSG00000004059        ARF5
2 ENST00000000412 ENSG00000003056        M6PR
3 ENST00000000442 ENSG00000173153       ESRRA
4 ENST00000001008 ENSG00000004478       FKBP4
5 ENST00000001146 ENSG00000003137     CYP26B1
6 ENST00000002125 ENSG00000003509     NDUFAF7</code></pre>
<pre class="r"><code>tx2gene&lt;- Tx.ensemble[,c(1,3)]

head(tx2gene)</code></pre>
<pre><code>DataFrame with 6 rows and 2 columns
            tx_id   gene_name
      &lt;character&gt; &lt;character&gt;
1 ENST00000000233        ARF5
2 ENST00000000412        M6PR
3 ENST00000000442       ESRRA
4 ENST00000001008       FKBP4
5 ENST00000001146     CYP26B1
6 ENST00000002125     NDUFAF7</code></pre>
</div>
<div id="create-a-deseqdataset-from-txi-count-table" class="section level1">
<h1>Create a DEseqDataSet from txi count table</h1>
<pre class="r"><code># Open file name

clinical_sample_info &lt;- read.csv(&quot;~/Dropbox/Lauren Blake/RNA_seq_project/clinical_sample_info.csv&quot;, stringsAsFactors = TRUE)
clinical_sample &lt;- as.data.frame(clinical_sample_info[,2:3], stringsAsFactors = TRUE)
clinical_sample[,1] &lt;- as.factor(clinical_sample[,1])
clinical_sample[,2] &lt;- as.factor(clinical_sample[,2])</code></pre>
</div>
<div id="check-hg38" class="section level1">
<h1>Check HG38</h1>
<pre class="r"><code>#source(&quot;https://bioconductor.org/biocLite.R&quot;)
#biocLite(&quot;EnsDb.Hsapiens.v86&quot;)

#create a tx2gene.txt table
#library(EnsDb.Hsapiens.v86)
#edb &lt;- EnsDb.Hsapiens.v86

#Tx.ensemble &lt;- transcripts(edb,
#          columns = c(&quot;tx_id&quot;, &quot;gene_id&quot;, &quot;gene_name&quot;),
#          return.type = &quot;DataFrame&quot;)
#nrow(Tx.ensemble)

#tx2gene_check&lt;- Tx.ensemble[,c(1:3)]
#dim(tx2gene_check)</code></pre>
</div>
<div id="import-quant.sf-data" class="section level1">
<h1>Import quant.sf data</h1>
<pre class="r"><code>#quant &lt;- read.delim(&quot;../data/quant.sf&quot;)</code></pre>
</div>
<div id="run-tximport" class="section level1">
<h1>Run tximport</h1>
<pre class="r"><code>#samples&lt;- c(&quot;trimmed_transcripts_quant_405-BAN22201T1-MH0163673_AATCCAGC_S9_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22201T2-MH0163857_CGTCTAAC_S3_L004_R1_001&quot;)
#samples.dir &lt;- c(&quot;/Users/laurenblake/Desktop/weight_restoration/weight_restoration/data&quot;)
#salmon.files&lt;- file.path(samples.dir, samples, &quot;quant.sf&quot;)
#names(salmon.files)&lt;- c(&quot;405-BAN22201T1-MH0163673_AATCCAGC_S9_L001_R1_001&quot;)
  
#all(file.exists(salmon.files))

#tx.salmon &lt;- tximport(salmon.files, type = &quot;salmon&quot;, tx2gene = tx2gene, dropInfReps=TRUE)

#salmon.counts&lt;- tx.salmon$counts
#salmon.counts&lt;- as.data.frame(salmon.counts)</code></pre>
</div>
<div id="run-for-all-files" class="section level1">
<h1>Run for all files</h1>
<pre class="r"><code>#files_list &lt;- read.table(&quot;../data/files_list.txt&quot;, quote=&quot;\&quot;&quot;, comment.char=&quot;&quot;, stringsAsFactors = FALSE)
#samples&lt;- unlist(files_list)
#samples&lt;- c(&quot;trimmed_transcripts_quant_405-BAN22201T1-MH0163673_AATCCAGC_S9_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22201T2-MH0163857_CGTCTAAC_S3_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22201T3-MH0167352_CTAAGACC_S31_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22202T1-MH0163700_CTTCCTTC_S19_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22202T2-MH0164426_GCACACAA_S4_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22202T3-MH0167384_ACTCCTAC_S18_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22203T1-MH0163897_TCCATTGC_S32_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22203T2-MH0164104_CTAAGACC_S21_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22204T1-MH0164041_CTTCCTTC_S19_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22204T2-MH0165543_CTCCTAGT_S8_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22204T3-MH0168921_AACTTGCC_S57_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22205T1-MH0164218_CGCTACAT_S7_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22205T2-MH0165296_AACTTGCC_S17_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22206T1-MH0164595_CGCTACAT_S7_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22206T2-MH0165241_GTCGAGAA_S6_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22206T3-MH0167362_CTCCTAGT_S68_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22207T1-MH0164655_ACAACAGC_S10_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22207T2-MH0165541_CCTTAGGT_S22_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22207T3-MH0168576_CGCTACAT_S75_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22208T1-MH0165002_CTCGAACA_S30_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22208T2-MH0165266_ATCGTCTC_S29_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22208T3-MH0168126_ACAACAGC_S48_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22209T1-MH0165265_AACTCGGA_S2_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22209T2-MH0165750_TCTAGGAG_S28_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22209T3-MH0168523_CGTCCATT_S16_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22210T1-MH0165547_AACCGAAC_S27_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22210T2-MH0166544_AACGCCTT_S26_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22210T3-MH0168587_CTTCCTTC_S56_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22212T1-MH0165601_GCACACAA_S4_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22212T2-MH0165753_CGTCCATT_S16_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22215T1-MH0165895_GACTACGA_S15_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22215T2-MH0166334_AATCCAGC_S9_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22215T3-MH0168486_CCTTAGGT_S32_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22216T1-MH0165949_CAAGCCAA_S31_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22216T2-MH0167360_GACTACGA_S15_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22216T3-MH0169379_CGTCTAAC_S47_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22218T1-MH0166539_GTACACCT_S14_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22218T2-MH0167895_CGACCTAA_S13_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22218T3-MH0169739_AATCCAGC_S14_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22219T1-MH0167403_CGTCTAAC_S3_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22219T2-MH0168012_ACTCCTAC_S18_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22219T3-MH0170053_CGTCCATT_S55_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22220T1-MH0167615_CCTATACC_S25_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22220T2-MH0168005_CCAACACT_S24_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22220T3-MH0169956_AACTCGGA_S2_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22221T1-MH0168112_ACCATCCT_S11_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22221T2-MH0168529_TACATCGG_S33_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22221T3-MH0171129_GTCGAGAA_S46_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22222T1-MH0168671_CTAAGACC_S21_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22222T2-MH0169515_TCTTCGAC_S74_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22224T1-MH0169098_CTTCCTTC_S73_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22224T2-MH0169571_GACTACGA_S21_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22224T3-MH0172869_AACCGAAC_S34_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22226T1-MH0169516_AACTTGCC_S22_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22226T2-MH0170691_ACGAGAAC_S23_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22226T3-MH0173165_GCACACAA_S24_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22228T1-MH0169654_CGTCCATT_S29_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22228T2-MH0171900_ATGACAGG_S28_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22228T3-MH0174229_ACCATCCT_S27_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22229T1-MH0170475_CTCCTAGT_S26_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22229T2-MH0171083_GTCGAGAA_S13_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22232T1-MH0171809_AACTCGGA_S76_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22232T2-MH0171901_GACTACGA_S15_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22232T3-MH0174220_ATCGTCTC_S71_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22233T1-MH0171922_GTACACCT_S4_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22233T2-MH0172715_GTACACCT_S30_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22233T3-MH0175373_ACGGACTT_S35_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22234T1-MH0172376_ACAACAGC_S10_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22234T2-MH0174414_CGCTACAT_S12_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22234T3-MH0177050_AACTCGGA_S49_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22235T1-MH0172813_TCCATTGC_S11_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22235T2-MH0173158_CGTCTAAC_S70_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22235T3-MH0175276_AACGCCTT_S36_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22236T1-MH0173201_CGACCTAA_S37_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22236T2-MH0174188_ACAACAGC_S15_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22239T1-MH0174265_ACTCCTAC_S25_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22239T2-MH0174778_TACATCGG_S12_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22239T3-MH0178097_CCTATACC_S38_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22240T1-MH0174969_CGCTACAT_S7_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22240T2-MH0175802_AACTTGCC_S17_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22242T1-MH0175396_ATGACAGG_S5_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22242T2-MH0177006_ACTCCTAC_S18_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22242T3-MH0181773_CAAGCCAA_S44_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22243T1-MH0175481_CCTATACC_S20_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22243T2-MH0176559_ACGGACTT_S3_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22243T3-MH0179740_TCTAGGAG_S39_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22245T1-MH0177498_ATGACAGG_S5_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22245T2-MH0179482_CAAGCCAA_S31_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22247T1-MH0178383_TCTTCGAC_S1_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22247T2-MH0179693_ACCATCCT_S11_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22248T1-MH0179252_CCAACACT_S24_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22248T2-MH0179936_ACGGACTT_S23_L003_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22248T3-MH0184420_CCAACACT_S69_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22249T1-MH0179995_TCTTCGAC_S1_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22249T2-MH0180995_CTCGAACA_S2_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22250T1-MH0180438_GTCGAGAA_S6_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22250T2-MH0181264_GCACACAA_S4_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22251T1-MH0181000_AACTCGGA_S2_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22251T2-MH0181244_CTCGAACA_S30_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22252T1-MH0181193_AACCGAAC_S19_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22252T2-MH0181540_ATGACAGG_S5_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22253T1-MH0181369_TACATCGG_S1_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22253T2-MH0183318_CTCCTAGT_S8_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22254T1-MH0183287_CCAACACT_S10_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22254T2-MH0183347_CCTTAGGT_S22_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22254T3-MH0186085_AATCCAGC_S72_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22255T1-MH0186017_CCTTAGGT_S18_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22255T2-MH0186145_GTCGAGAA_S6_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22256T1-MH0186190_ACGAGAAC_S20_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22256T2-MH0186757_CGTCCATT_S16_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22256T3-MH0192924_TCCATTGC_S43_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22257T1-MH0186793_CTCCTAGT_S8_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22257T2-MH0187462_TCCATTGC_S32_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22258T1-MH0187064_AATCCAGC_S9_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22258T2-MH0187605_CTAAGACC_S6_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22258T3-MH0198483_ACCATCCT_S54_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22260T1-MH0187236_GTACACCT_S14_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22260T2-MH0188240_CTTCCTTC_S19_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22260T3-MH0198647_ATGACAGG_S42_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22261T1-MH0188182_ACGGACTT_S23_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22261T2-MH0191258_CGACCTAA_S7_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22262T1-MH0188248_TCTTCGAC_S1_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22262T2-MH0191318_CCTATACC_S25_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22266T1-MH0192891_ACAACAGC_S10_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22266T2-MH0197478_ACGAGAAC_S20_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22266T3-MH0201141_GACTACGA_S53_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22267T1-MH0197522_AACTTGCC_S17_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22267T2-MH0197979_ATCGTCTC_S29_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22268T1-MH0197785_TCTAGGAG_S28_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22268T2-MH0197998_CGACCTAA_S13_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22268T3-MH0203576_ACTCCTAC_S52_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22269T1-MH0197823_CGTCTAAC_S3_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22269T2-MH0197945_ATCGTCTC_S8_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22270T1-MH0199425_CAAGCCAA_S17_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22270T2-MH0199506_ACCATCCT_S11_L008_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22271T1-MH0200966_AACGCCTT_S26_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22271T2-MH0201765_AACCGAAC_S27_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22271T3-MH0217142_CTCGAACA_S40_L004_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22272T1-MH0201784_TACATCGG_S12_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22272T2-MH0202184_AACGCCTT_S16_L002_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22272T3-MH0217144_GCACACAA_S41_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22274T1-MH0201904_TCTAGGAG_S9_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22274T2-MH0202323_ACGAGAAC_S5_L001_R1_001&quot;, &quot;trimmed_transcripts_quant_405-BAN22274T3-MH0217163_TCTTCGAC_S45_L005_R1_001&quot;, &quot;trimmed_transcripts_quant_BAK_BSP_001-BAN22202_RRED397T1_GTACACCT_S51_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_BAK_BSP_002-BAN22202_RRED397T2_ACGAGAAC_S50_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_BAK_BSP_003-BAN22220_RRED398T1_CGACCTAA_S58_L006_R1_001&quot;, &quot;trimmed_transcripts_quant_BAK_BSP_004-BAN22220_RRED398T2_TACATCGG_S59_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_BAK_BSP_005-BAN22209_RRED399T1_ATCGTCTC_S60_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_BAK_BSP_006-BAN22209_RRED399T2_CCAACACT_S67_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_BAK_BSP_007-BAN22218_RRED424T1_TCTAGGAG_S66_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_BAK_BSP_008-BAN22218_RRED424T2_CTCGAACA_S65_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_BAK_BSP_009-BAN22226_RRED453T1_ACGGACTT_S64_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_BAK_BSP_010-BAN22226_RRED453T2_CTAAGACC_S63_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_BAK_BSP_011-BAN22228_RRED467T1_AACCGAAC_S62_L007_R1_001&quot;, &quot;trimmed_transcripts_quant_BAK_BSP_012-BAN22228_RRED467T2_CCTTAGGT_S61_L007_R1_001&quot;)

samples&lt;- c(&quot;405-BAN22201T1-MH0163673_AATCCAGC_S9_L001_R1_001&quot;,
&quot;405-BAN22201T2-MH0163857_CGTCTAAC_S3_L004_R1_001&quot;,
&quot;405-BAN22201T3-MH0167352_CTAAGACC_S31_L004_R1_001&quot;,
&quot;405-BAN22202T1-MH0163700_CTTCCTTC_S19_L002_R1_001&quot;,
&quot;405-BAN22202T2-MH0164426_GCACACAA_S4_L004_R1_001&quot;,
&quot;405-BAN22202T3-MH0167384_ACTCCTAC_S18_L002_R1_001&quot;,
&quot;405-BAN22203T1-MH0163897_TCCATTGC_S32_L004_R1_001&quot;,
&quot;405-BAN22203T2-MH0164104_CTAAGACC_S21_L003_R1_001&quot;,
&quot;405-BAN22204T1-MH0164041_CTTCCTTC_S19_L005_R1_001&quot;,
&quot;405-BAN22204T2-MH0165543_CTCCTAGT_S8_L001_R1_001&quot;,
&quot;405-BAN22204T3-MH0168921_AACTTGCC_S57_L006_R1_001&quot;,
&quot;405-BAN22205T1-MH0164218_CGCTACAT_S7_L004_R1_001&quot;,
&quot;405-BAN22205T2-MH0165296_AACTTGCC_S17_L002_R1_001&quot;,
&quot;405-BAN22206T1-MH0164595_CGCTACAT_S7_L001_R1_001&quot;,
&quot;405-BAN22206T2-MH0165241_GTCGAGAA_S6_L001_R1_001&quot;,
&quot;405-BAN22206T3-MH0167362_CTCCTAGT_S68_L008_R1_001&quot;,
&quot;405-BAN22207T1-MH0164655_ACAACAGC_S10_L001_R1_001&quot;,
&quot;405-BAN22207T2-MH0165541_CCTTAGGT_S22_L003_R1_001&quot;,
&quot;405-BAN22207T3-MH0168576_CGCTACAT_S75_L008_R1_001&quot;,
&quot;405-BAN22208T1-MH0165002_CTCGAACA_S30_L003_R1_001&quot;,
&quot;405-BAN22208T2-MH0165266_ATCGTCTC_S29_L003_R1_001&quot;,
&quot;405-BAN22208T3-MH0168126_ACAACAGC_S48_L005_R1_001&quot;,
&quot;405-BAN22209T1-MH0165265_AACTCGGA_S2_L004_R1_001&quot;,
&quot;405-BAN22209T2-MH0165750_TCTAGGAG_S28_L003_R1_001&quot;,
&quot;405-BAN22209T3-MH0168523_CGTCCATT_S16_L005_R1_001&quot;,
&quot;405-BAN22210T1-MH0165547_AACCGAAC_S27_L003_R1_001&quot;,
&quot;405-BAN22210T2-MH0166544_AACGCCTT_S26_L003_R1_001&quot;,
&quot;405-BAN22210T3-MH0168587_CTTCCTTC_S56_L006_R1_001&quot;,
&quot;405-BAN22212T1-MH0165601_GCACACAA_S4_L001_R1_001&quot;,
&quot;405-BAN22212T2-MH0165753_CGTCCATT_S16_L002_R1_001&quot;,
&quot;405-BAN22215T1-MH0165895_GACTACGA_S15_L005_R1_001&quot;,
&quot;405-BAN22215T2-MH0166334_AATCCAGC_S9_L004_R1_001&quot;,
&quot;405-BAN22215T3-MH0168486_CCTTAGGT_S32_L004_R1_001&quot;,
&quot;405-BAN22216T1-MH0165949_CAAGCCAA_S31_L004_R1_001&quot;,
&quot;405-BAN22216T2-MH0167360_GACTACGA_S15_L002_R1_001&quot;,
&quot;405-BAN22216T3-MH0169379_CGTCTAAC_S47_L005_R1_001&quot;,
&quot;405-BAN22218T1-MH0166539_GTACACCT_S14_L002_R1_001&quot;,
&quot;405-BAN22218T2-MH0167895_CGACCTAA_S13_L002_R1_001&quot;,
&quot;405-BAN22218T3-MH0169739_AATCCAGC_S14_L002_R1_001&quot;,
&quot;405-BAN22219T1-MH0167403_CGTCTAAC_S3_L001_R1_001&quot;,
&quot;405-BAN22219T2-MH0168012_ACTCCTAC_S18_L005_R1_001&quot;,
&quot;405-BAN22219T3-MH0170053_CGTCCATT_S55_L006_R1_001&quot;,
&quot;405-BAN22220T1-MH0167615_CCTATACC_S25_L003_R1_001&quot;,
&quot;405-BAN22220T2-MH0168005_CCAACACT_S24_L003_R1_001&quot;,
&quot;405-BAN22220T3-MH0169956_AACTCGGA_S2_L001_R1_001&quot;,
&quot;405-BAN22221T1-MH0168112_ACCATCCT_S11_L005_R1_001&quot;,
&quot;405-BAN22221T2-MH0168529_TACATCGG_S33_L004_R1_001&quot;,
&quot;405-BAN22221T3-MH0171129_GTCGAGAA_S46_L005_R1_001&quot;,
&quot;405-BAN22222T1-MH0168671_CTAAGACC_S21_L006_R1_001&quot;,
&quot;405-BAN22222T2-MH0169515_TCTTCGAC_S74_L008_R1_001&quot;,
&quot;405-BAN22224T1-MH0169098_CTTCCTTC_S73_L008_R1_001&quot;,
&quot;405-BAN22224T2-MH0169571_GACTACGA_S21_L003_R1_001&quot;,
&quot;405-BAN22224T3-MH0172869_AACCGAAC_S34_L004_R1_001&quot;,
&quot;405-BAN22226T1-MH0169516_AACTTGCC_S22_L003_R1_001&quot;,
&quot;405-BAN22226T2-MH0170691_ACGAGAAC_S23_L003_R1_001&quot;,
&quot;405-BAN22226T3-MH0173165_GCACACAA_S24_L003_R1_001&quot;,
&quot;405-BAN22228T1-MH0169654_CGTCCATT_S29_L003_R1_001&quot;,
&quot;405-BAN22228T2-MH0171900_ATGACAGG_S28_L003_R1_001&quot;,
&quot;405-BAN22228T3-MH0174229_ACCATCCT_S27_L003_R1_001&quot;,
&quot;405-BAN22229T1-MH0170475_CTCCTAGT_S26_L003_R1_001&quot;,
&quot;405-BAN22229T2-MH0171083_GTCGAGAA_S13_L002_R1_001&quot;,
&quot;405-BAN22232T1-MH0171809_AACTCGGA_S76_L008_R1_001&quot;,
&quot;405-BAN22232T2-MH0171901_GACTACGA_S15_L008_R1_001&quot;,
&quot;405-BAN22232T3-MH0174220_ATCGTCTC_S71_L008_R1_001&quot;,
&quot;405-BAN22233T1-MH0171922_GTACACCT_S4_L001_R1_001&quot;,
&quot;405-BAN22233T2-MH0172715_GTACACCT_S30_L003_R1_001&quot;,
&quot;405-BAN22233T3-MH0175373_ACGGACTT_S35_L004_R1_001&quot;,
&quot;405-BAN22234T1-MH0172376_ACAACAGC_S10_L007_R1_001&quot;,
&quot;405-BAN22234T2-MH0174414_CGCTACAT_S12_L002_R1_001&quot;,
&quot;405-BAN22234T3-MH0177050_AACTCGGA_S49_L005_R1_001&quot;,
&quot;405-BAN22235T1-MH0172813_TCCATTGC_S11_L002_R1_001&quot;,
&quot;405-BAN22235T2-MH0173158_CGTCTAAC_S70_L008_R1_001&quot;,
&quot;405-BAN22235T3-MH0175276_AACGCCTT_S36_L004_R1_001&quot;,
&quot;405-BAN22236T1-MH0173201_CGACCTAA_S37_L004_R1_001&quot;,
&quot;405-BAN22236T2-MH0174188_ACAACAGC_S15_L002_R1_001&quot;,
&quot;405-BAN22239T1-MH0174265_ACTCCTAC_S25_L003_R1_001&quot;,
&quot;405-BAN22239T2-MH0174778_TACATCGG_S12_L002_R1_001&quot;,
&quot;405-BAN22239T3-MH0178097_CCTATACC_S38_L004_R1_001&quot;,
&quot;405-BAN22240T1-MH0174969_CGCTACAT_S7_L007_R1_001&quot;,
&quot;405-BAN22240T2-MH0175802_AACTTGCC_S17_L008_R1_001&quot;,
&quot;405-BAN22242T1-MH0175396_ATGACAGG_S5_L008_R1_001&quot;,
&quot;405-BAN22242T2-MH0177006_ACTCCTAC_S18_L008_R1_001&quot;,
&quot;405-BAN22242T3-MH0181773_CAAGCCAA_S44_L005_R1_001&quot;,
&quot;405-BAN22243T1-MH0175481_CCTATACC_S20_L002_R1_001&quot;,
&quot;405-BAN22243T2-MH0176559_ACGGACTT_S3_L001_R1_001&quot;,
&quot;405-BAN22243T3-MH0179740_TCTAGGAG_S39_L004_R1_001&quot;,
&quot;405-BAN22245T1-MH0177498_ATGACAGG_S5_L004_R1_001&quot;,
&quot;405-BAN22245T2-MH0179482_CAAGCCAA_S31_L007_R1_001&quot;,
&quot;405-BAN22247T1-MH0178383_TCTTCGAC_S1_L001_R1_001&quot;,
&quot;405-BAN22247T2-MH0179693_ACCATCCT_S11_L002_R1_001&quot;,
&quot;405-BAN22248T1-MH0179252_CCAACACT_S24_L006_R1_001&quot;,
&quot;405-BAN22248T2-MH0179936_ACGGACTT_S23_L003_R1_001&quot;,
&quot;405-BAN22248T3-MH0184420_CCAACACT_S69_L008_R1_001&quot;,
&quot;405-BAN22249T1-MH0179995_TCTTCGAC_S1_L005_R1_001&quot;,
&quot;405-BAN22249T2-MH0180995_CTCGAACA_S2_L001_R1_001&quot;,
&quot;405-BAN22250T1-MH0180438_GTCGAGAA_S6_L004_R1_001&quot;,
&quot;405-BAN22250T2-MH0181264_GCACACAA_S4_L008_R1_001&quot;,
&quot;405-BAN22251T1-MH0181000_AACTCGGA_S2_L007_R1_001&quot;,
&quot;405-BAN22251T2-MH0181244_CTCGAACA_S30_L006_R1_001&quot;,
&quot;405-BAN22252T1-MH0181193_AACCGAAC_S19_L002_R1_001&quot;,
&quot;405-BAN22252T2-MH0181540_ATGACAGG_S5_L001_R1_001&quot;,
&quot;405-BAN22253T1-MH0181369_TACATCGG_S1_L001_R1_001&quot;,
&quot;405-BAN22253T2-MH0183318_CTCCTAGT_S8_L005_R1_001&quot;,
&quot;405-BAN22254T1-MH0183287_CCAACACT_S10_L001_R1_001&quot;,
&quot;405-BAN22254T2-MH0183347_CCTTAGGT_S22_L006_R1_001&quot;,
&quot;405-BAN22254T3-MH0186085_AATCCAGC_S72_L008_R1_001&quot;,
&quot;405-BAN22255T1-MH0186017_CCTTAGGT_S18_L002_R1_001&quot;,
&quot;405-BAN22255T2-MH0186145_GTCGAGAA_S6_L007_R1_001&quot;,
&quot;405-BAN22256T1-MH0186190_ACGAGAAC_S20_L005_R1_001&quot;,
&quot;405-BAN22256T2-MH0186757_CGTCCATT_S16_L008_R1_001&quot;,
&quot;405-BAN22256T3-MH0192924_TCCATTGC_S43_L005_R1_001&quot;,
&quot;405-BAN22257T1-MH0186793_CTCCTAGT_S8_L008_R1_001&quot;,
&quot;405-BAN22257T2-MH0187462_TCCATTGC_S32_L007_R1_001&quot;,
&quot;405-BAN22258T1-MH0187064_AATCCAGC_S9_L007_R1_001&quot;,
&quot;405-BAN22258T2-MH0187605_CTAAGACC_S6_L001_R1_001&quot;,
&quot;405-BAN22258T3-MH0198483_ACCATCCT_S54_L006_R1_001&quot;,
&quot;405-BAN22260T1-MH0187236_GTACACCT_S14_L005_R1_001&quot;,
&quot;405-BAN22260T2-MH0188240_CTTCCTTC_S19_L008_R1_001&quot;,
&quot;405-BAN22260T3-MH0198647_ATGACAGG_S42_L005_R1_001&quot;,
&quot;405-BAN22261T1-MH0188182_ACGGACTT_S23_L006_R1_001&quot;,
&quot;405-BAN22261T2-MH0191258_CGACCTAA_S7_L001_R1_001&quot;,
&quot;405-BAN22262T1-MH0188248_TCTTCGAC_S1_L008_R1_001&quot;,
&quot;405-BAN22262T2-MH0191318_CCTATACC_S25_L007_R1_001&quot;,
&quot;405-BAN22266T1-MH0192891_ACAACAGC_S10_L004_R1_001&quot;,
&quot;405-BAN22266T2-MH0197478_ACGAGAAC_S20_L002_R1_001&quot;,
&quot;405-BAN22266T3-MH0201141_GACTACGA_S53_L006_R1_001&quot;,
&quot;405-BAN22267T1-MH0197522_AACTTGCC_S17_L005_R1_001&quot;,
&quot;405-BAN22267T2-MH0197979_ATCGTCTC_S29_L006_R1_001&quot;,
&quot;405-BAN22268T1-MH0197785_TCTAGGAG_S28_L006_R1_001&quot;,
&quot;405-BAN22268T2-MH0197998_CGACCTAA_S13_L006_R1_001&quot;,
&quot;405-BAN22268T3-MH0203576_ACTCCTAC_S52_L006_R1_001&quot;,
&quot;405-BAN22269T1-MH0197823_CGTCTAAC_S3_L007_R1_001&quot;,
&quot;405-BAN22269T2-MH0197945_ATCGTCTC_S8_L001_R1_001&quot;,
&quot;405-BAN22270T1-MH0199425_CAAGCCAA_S17_L002_R1_001&quot;,
&quot;405-BAN22270T2-MH0199506_ACCATCCT_S11_L008_R1_001&quot;,
&quot;405-BAN22271T1-MH0200966_AACGCCTT_S26_L007_R1_001&quot;,
&quot;405-BAN22271T2-MH0201765_AACCGAAC_S27_L006_R1_001&quot;,
&quot;405-BAN22271T3-MH0217142_CTCGAACA_S40_L004_R1_001&quot;,
&quot;405-BAN22272T1-MH0201784_TACATCGG_S12_L006_R1_001&quot;,
&quot;405-BAN22272T2-MH0202184_AACGCCTT_S16_L002_R1_001&quot;,
&quot;405-BAN22272T3-MH0217144_GCACACAA_S41_L005_R1_001&quot;,
&quot;405-BAN22274T1-MH0201904_TCTAGGAG_S9_L001_R1_001&quot;,
&quot;405-BAN22274T2-MH0202323_ACGAGAAC_S5_L001_R1_001&quot;,
&quot;405-BAN22274T3-MH0217163_TCTTCGAC_S45_L005_R1_001&quot;,
&quot;BAK_BSP_001-BAN22202_RRED397T1_GTACACCT_S51_L006_R1_001&quot;,
&quot;BAK_BSP_002-BAN22202_RRED397T2_ACGAGAAC_S50_L006_R1_001&quot;,
&quot;BAK_BSP_003-BAN22220_RRED398T1_CGACCTAA_S58_L006_R1_001&quot;,
&quot;BAK_BSP_004-BAN22220_RRED398T2_TACATCGG_S59_L007_R1_001&quot;,
&quot;BAK_BSP_005-BAN22209_RRED399T1_ATCGTCTC_S60_L007_R1_001&quot;,
&quot;BAK_BSP_006-BAN22209_RRED399T2_CCAACACT_S67_L007_R1_001&quot;,
&quot;BAK_BSP_007-BAN22218_RRED424T1_TCTAGGAG_S66_L007_R1_001&quot;,
&quot;BAK_BSP_008-BAN22218_RRED424T2_CTCGAACA_S65_L007_R1_001&quot;,
&quot;BAK_BSP_009-BAN22226_RRED453T1_ACGGACTT_S64_L007_R1_001&quot;,
&quot;BAK_BSP_010-BAN22226_RRED453T2_CTAAGACC_S63_L007_R1_001&quot;,
&quot;BAK_BSP_011-BAN22228_RRED467T1_AACCGAAC_S62_L007_R1_001&quot;,
&quot;BAK_BSP_012-BAN22228_RRED467T2_CCTTAGGT_S61_L007_R1_001&quot;)

samples.dir &lt;- c(&quot;/Users/laurenblake/Desktop/weight_restoration/weight_restoration/data&quot;)
salmon.files&lt;- file.path(samples.dir, samples, &quot;quant.sf&quot;)
names(salmon.files)&lt;-samples&lt;- c(&quot;BAN22201T1&quot;,
&quot;BAN22201T2&quot;,
&quot;BAN22201T3&quot;,
&quot;BAN22202T1&quot;,
&quot;BAN22202T2&quot;,
&quot;BAN22202T3&quot;,
&quot;BAN22203T1&quot;,
&quot;BAN22203T2&quot;,
&quot;BAN22204T1&quot;,
&quot;BAN22204T2&quot;,
&quot;BAN22204T3&quot;,
&quot;BAN22205T1&quot;,
&quot;BAN22205T2&quot;,
&quot;BAN22206T1&quot;,
&quot;BAN22206T2&quot;,
&quot;BAN22206T3&quot;,
&quot;BAN22207T1&quot;,
&quot;BAN22207T2&quot;,
&quot;BAN22207T3&quot;,
&quot;BAN22208T1&quot;,
&quot;BAN22208T2&quot;,
&quot;BAN22208T3&quot;,
&quot;BAN22209T1&quot;,
&quot;BAN22209T2&quot;,
&quot;BAN22209T3&quot;,
&quot;BAN22210T1&quot;,
&quot;BAN22210T2&quot;,
&quot;BAN22210T3&quot;,
&quot;BAN22212T1&quot;,
&quot;BAN22212T2&quot;,
&quot;BAN22215T1&quot;,
&quot;BAN22215T2&quot;,
&quot;BAN22215T3&quot;,
&quot;BAN22216T1&quot;,
&quot;BAN22216T2&quot;,
&quot;BAN22216T3&quot;,
&quot;BAN22218T1&quot;,
&quot;BAN22218T2&quot;,
&quot;BAN22218T3&quot;,
&quot;BAN22219T1&quot;,
&quot;BAN22219T2&quot;,
&quot;BAN22219T3&quot;,
&quot;BAN22220T1&quot;,
&quot;BAN22220T2&quot;,
&quot;BAN22220T3&quot;,
&quot;BAN22221T1&quot;,
&quot;BAN22221T2&quot;,
&quot;BAN22221T3&quot;,
&quot;BAN22222T1&quot;,
&quot;BAN22222T2&quot;,
&quot;BAN22224T1&quot;,
&quot;BAN22224T2&quot;,
&quot;BAN22224T3&quot;,
&quot;BAN22226T1&quot;,
&quot;BAN22226T2&quot;,
&quot;BAN22226T3&quot;,
&quot;BAN22228T1&quot;,
&quot;BAN22228T2&quot;,
&quot;BAN22228T3&quot;,
&quot;BAN22229T1&quot;,
&quot;BAN22229T2&quot;,
&quot;BAN22232T1&quot;,
&quot;BAN22232T2&quot;,
&quot;BAN22232T3&quot;,
&quot;BAN22233T1&quot;,
&quot;BAN22233T2&quot;,
&quot;BAN22233T3&quot;,
&quot;BAN22234T1&quot;,
&quot;BAN22234T2&quot;,
&quot;BAN22234T3&quot;,
&quot;BAN22235T1&quot;,
&quot;BAN22235T2&quot;,
&quot;BAN22235T3&quot;,
&quot;BAN22236T1&quot;,
&quot;BAN22236T2&quot;,
&quot;BAN22239T1&quot;,
&quot;BAN22239T2&quot;,
&quot;BAN22239T3&quot;,
&quot;BAN22240T1&quot;,
&quot;BAN22240T2&quot;,
&quot;BAN22242T1&quot;,
&quot;BAN22242T2&quot;,
&quot;BAN22242T3&quot;,
&quot;BAN22243T1&quot;,
&quot;BAN22243T2&quot;,
&quot;BAN22243T3&quot;,
&quot;BAN22245T1&quot;,
&quot;BAN22245T2&quot;,
&quot;BAN22247T1&quot;,
&quot;BAN22247T2&quot;,
&quot;BAN22248T1&quot;,
&quot;BAN22248T2&quot;,
&quot;BAN22248T3&quot;,
&quot;BAN22249T1&quot;,
&quot;BAN22249T2&quot;,
&quot;BAN22250T1&quot;,
&quot;BAN22250T2&quot;,
&quot;BAN22251T1&quot;,
&quot;BAN22251T2&quot;,
&quot;BAN22252T1&quot;,
&quot;BAN22252T2&quot;,
&quot;BAN22253T1&quot;,
&quot;BAN22253T2&quot;,
&quot;BAN22254T1&quot;,
&quot;BAN22254T2&quot;,
&quot;BAN22254T3&quot;,
&quot;BAN22255T1&quot;,
&quot;BAN22255T2&quot;,
&quot;BAN22256T1&quot;,
&quot;BAN22256T2&quot;,
&quot;BAN22256T3&quot;,
&quot;BAN22257T1&quot;,
&quot;BAN22257T2&quot;,
&quot;BAN22258T1&quot;,
&quot;BAN22258T2&quot;,
&quot;BAN22258T3&quot;,
&quot;BAN22260T1&quot;,
&quot;BAN22260T2&quot;,
&quot;BAN22260T3&quot;,
&quot;BAN22261T1&quot;,
&quot;BAN22261T2&quot;,
&quot;BAN22262T1&quot;,
&quot;BAN22262T2&quot;,
&quot;BAN22266T1&quot;,
&quot;BAN22266T2&quot;,
&quot;BAN22266T3&quot;,
&quot;BAN22267T1&quot;,
&quot;BAN22267T2&quot;,
&quot;BAN22268T1&quot;,
&quot;BAN22268T2&quot;,
&quot;BAN22268T3&quot;,
&quot;BAN22269T1&quot;,
&quot;BAN22269T2&quot;,
&quot;BAN22270T1&quot;,
&quot;BAN22270T2&quot;,
&quot;BAN22271T1&quot;,
&quot;BAN22271T2&quot;,
&quot;BAN22271T3&quot;,
&quot;BAN22272T1&quot;,
&quot;BAN22272T2&quot;,
&quot;BAN22272T3&quot;,
&quot;BAN22274T1&quot;,
&quot;BAN22274T2&quot;,
&quot;BAN22274T3&quot;,
&quot;BAN22202_RRED397T1&quot;,
&quot;BAN22202_RRED397T2&quot;,
&quot;BAN22220_RRED398T1&quot;,
&quot;BAN22220_RRED398T2&quot;,
&quot;BAN22209_RRED399T1&quot;,
&quot;BAN22209_RRED399T2&quot;,
&quot;BAN22218_RRED424T1&quot;,
&quot;BAN22218_RRED424T2&quot;,
&quot;BAN22226_RRED453T1&quot;,
&quot;BAN22226_RRED453T2&quot;,
&quot;BAN22228_RRED467T1&quot;,
&quot;BAN22228_RRED467T2&quot;)

  
all(file.exists(salmon.files))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code> #tx2gene &lt;-  tx2gene[which( tx2gene$gene_name != &quot;HBB&quot; ),]
 #tx2gene &lt;-  tx2gene[which( tx2gene$gene_name != &quot;HBA2&quot; ),]
 #tx2gene &lt;-  tx2gene[which( tx2gene$gene_name != &quot;HBA1&quot; ),]

 
tx.salmon &lt;- tximport(salmon.files, type = &quot;salmon&quot;, tx2gene = tx2gene, dropInfReps=TRUE)</code></pre>
<pre><code>reading in files with read_tsv</code></pre>
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 
summarizing abundance
summarizing counts
summarizing length</code></pre>
<pre class="r"><code>salmon.counts&lt;- tx.salmon$counts
salmon.counts&lt;- as.data.frame(salmon.counts)
check &lt;- salmon.counts[which(rownames(salmon.counts) == &quot;HBB&quot;),]

salmon_counts &lt;- as.data.frame(tx.salmon$counts)
plot(rowSums(salmon_counts))</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(rowSums(salmon_counts), pch = 19, ylab = &quot;Sum of estimated raw reads across all samples&quot;, xlab = &quot; &quot;, ylim=c(0, 303921608), main = &quot;Sum of estimated read counts for each gene (all samples)&quot;) </code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>salmon_counts1 &lt;-  salmon_counts[which( rownames(salmon_counts) != &quot;HBB&quot; ),]
salmon_counts1 &lt;-  salmon_counts1[which( rownames(salmon_counts) != &quot;HBA2&quot; ),]
salmon_counts1 &lt;-  salmon_counts1[which( rownames(salmon_counts) != &quot;HBA1&quot; ),]

plot(rowSums(salmon_counts1), ylab = &quot;Sum of estimated raw reads across all samples&quot;, xlab = &quot; &quot;, ylim=c(0, 303921608)) </code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>salmon.counts.hbb &lt;- salmon.counts[which(rownames(salmon.counts) == &quot;HBB&quot;),]
salmon.counts.hba1 &lt;- salmon.counts[which(rownames(salmon.counts) == &quot;HBA1&quot;),]
salmon.counts.hba2 &lt;- salmon.counts[which(rownames(salmon.counts) == &quot;HBA2&quot;),]


# HBB ENSG00000188170
# HBA2 ENSG00000188536
# HBA1 ENSG00000206172</code></pre>
</div>
<div id="remove-hba-then-run-vsd" class="section level1">
<h1>Remove HBA then run VSD</h1>
<pre class="r"><code># Remove 
# HBB ENSG00000188170
# HBA2 ENSG00000188536
# HBA1 ENSG00000206172




dds &lt;- DESeqDataSetFromTximport(tx.salmon, colData=clinical_sample, design = ~ BAN_ID + Time)</code></pre>
<pre><code>using counts and average transcript lengths from tximport</code></pre>
<pre class="r"><code>dds &lt;- dds[which(rownames(dds) != &quot;HBB&quot; ),]
dds &lt;- dds[which(rownames(dds) != &quot;HBA2&quot; ),]
dds &lt;- dds[which(rownames(dds) != &quot;HBA1&quot; ),]

vsd &lt;- vst(dds, blind = TRUE)</code></pre>
<pre><code>using &#39;avgTxLength&#39; from assays(dds), correcting for library size</code></pre>
<pre class="r"><code>X = t(scale(t(assay(vsd)),center=TRUE,scale=FALSE))
sv = svd(t(X))
U = sv$u
V = sv$v
D = sv$d
## in R calculate the rank of a matrix is by
#qr(t(X))$rank

plot(U[,1],U[,2],xlab=&quot;PC1&quot;,ylab=&quot;PC2&quot;)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cbPalette &lt;- c(&quot;#009E73&quot;, &quot;#E69F00&quot;, &quot;#999999&quot;, &quot;black&quot;, &quot;#56B4E9&quot;)

group.colors &lt;- c(&quot;1&quot; = &quot;#333BFF&quot;, &quot;2&quot; = &quot;#CC6600&quot;, &quot;3&quot; =&quot;#9633FF&quot;, &quot;4&quot; = &quot;#E2FF33&quot;, &quot;5&quot; = &quot;#E3DB71&quot;)


U &lt;- as.data.frame(U)
norm_count &lt;- ggplot(data=U, aes(x=U[,1], y=U[,2], color=factor(dds$Time))) + geom_point() + scale_color_discrete(name = &quot;Time&quot;) + xlab(&quot;PC1 (14% of variation explained)&quot;) + ylab(&quot;PC2 (8% of variation explained)&quot;) +  scale_colour_manual(name = &quot;Time&quot;, values=c(&quot;#F8766D&quot;, &quot;#B79F00&quot;, &quot;#00BA38&quot;, &quot;#3399FF&quot;, &quot;#CC66FF&quot;))</code></pre>
<pre><code>Scale for &#39;colour&#39; is already present. Adding another scale for
&#39;colour&#39;, which will replace the existing scale.</code></pre>
<pre class="r"><code>plot_grid(norm_count)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save_plot(&quot;/Users/laurenblake/Dropbox/Figures/PCA_time_no_globin.png&quot;, norm_count,
          base_aspect_ratio = 1, base_width = 6.95, base_height = 5.25)


U &lt;- as.data.frame(U)
norm_count &lt;- ggplot(data=U, aes(x=U[,1], y=U[,2], color=factor(dds$Time))) + geom_point() + scale_color_discrete(name = &quot;Time&quot;) + xlab(&quot;PC1 (14% of variation explained)&quot;) + ylab(&quot;PC2 (8% of variation explained)&quot;) +  scale_colour_manual(name = &quot;Time&quot;, values=c(&quot;#FFFFFF&quot;, &quot;#FFFFFF&quot;, &quot;#FFFFFF&quot;,  &quot;#3399FF&quot;, &quot;#CC66FF&quot;))</code></pre>
<pre><code>Scale for &#39;colour&#39; is already present. Adding another scale for
&#39;colour&#39;, which will replace the existing scale.</code></pre>
<pre class="r"><code>plot_grid(norm_count)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save_plot(&quot;/Users/laurenblake/Dropbox/Figures/PCA_time_no_globin_T5.png&quot;, norm_count,
          base_aspect_ratio = 1, base_width = 6.95, base_height = 5.25)

summary(lm(U[,1]~as.factor(dds$BAN_ID)))</code></pre>
<pre><code>
Call:
lm(formula = U[, 1] ~ as.factor(dds$BAN_ID))

Residuals:
      Min        1Q    Median        3Q       Max 
-0.152497 -0.043651 -0.000805  0.045545  0.138703 

Coefficients:
                            Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept)                0.0084715  0.0432692   0.196   0.8452  
as.factor(dds$BAN_ID)2202  0.0045927  0.0547317   0.084   0.9333  
as.factor(dds$BAN_ID)2203 -0.1263358  0.0684147  -1.847   0.0677 .
as.factor(dds$BAN_ID)2204 -0.0168967  0.0611919  -0.276   0.7830  
as.factor(dds$BAN_ID)2205 -0.1116596  0.0684147  -1.632   0.1058  
as.factor(dds$BAN_ID)2206 -0.1347549  0.0611919  -2.202   0.0299 *
as.factor(dds$BAN_ID)2207 -0.0735782  0.0611919  -1.202   0.2320  
as.factor(dds$BAN_ID)2208 -0.0118980  0.0611919  -0.194   0.8462  
as.factor(dds$BAN_ID)2209  0.0166058  0.0547317   0.303   0.7622  
as.factor(dds$BAN_ID)2210 -0.0565098  0.0611919  -0.923   0.3580  
as.factor(dds$BAN_ID)2212 -0.0816808  0.0684147  -1.194   0.2353  
as.factor(dds$BAN_ID)2215  0.0055438  0.0611919   0.091   0.9280  
as.factor(dds$BAN_ID)2216 -0.0579643  0.0611919  -0.947   0.3458  
as.factor(dds$BAN_ID)2218  0.0154135  0.0547317   0.282   0.7788  
as.factor(dds$BAN_ID)2219 -0.0328262  0.0611919  -0.536   0.5928  
as.factor(dds$BAN_ID)2220  0.0579615  0.0547317   1.059   0.2921  
as.factor(dds$BAN_ID)2221  0.0108536  0.0611919   0.177   0.8596  
as.factor(dds$BAN_ID)2222 -0.0520875  0.0684147  -0.761   0.4482  
as.factor(dds$BAN_ID)2224 -0.0088811  0.0611919  -0.145   0.8849  
as.factor(dds$BAN_ID)2226  0.0313737  0.0547317   0.573   0.5678  
as.factor(dds$BAN_ID)2228  0.0266684  0.0547317   0.487   0.6271  
as.factor(dds$BAN_ID)2229  0.0667968  0.0684147   0.976   0.3312  
as.factor(dds$BAN_ID)2232  0.0563843  0.0611919   0.921   0.3590  
as.factor(dds$BAN_ID)2233 -0.0355127  0.0611919  -0.580   0.5630  
as.factor(dds$BAN_ID)2234 -0.0268590  0.0611919  -0.439   0.6616  
as.factor(dds$BAN_ID)2235 -0.0801240  0.0611919  -1.309   0.1934  
as.factor(dds$BAN_ID)2236 -0.0345348  0.0684147  -0.505   0.6148  
as.factor(dds$BAN_ID)2239 -0.0549864  0.0611919  -0.899   0.3710  
as.factor(dds$BAN_ID)2240 -0.0041150  0.0684147  -0.060   0.9522  
as.factor(dds$BAN_ID)2242  0.0461569  0.0611919   0.754   0.4524  
as.factor(dds$BAN_ID)2243  0.0919864  0.0611919   1.503   0.1359  
as.factor(dds$BAN_ID)2245  0.0174058  0.0684147   0.254   0.7997  
as.factor(dds$BAN_ID)2247  0.0180506  0.0684147   0.264   0.7924  
as.factor(dds$BAN_ID)2248 -0.0087455  0.0611919  -0.143   0.8866  
as.factor(dds$BAN_ID)2249  0.0423033  0.0684147   0.618   0.5377  
as.factor(dds$BAN_ID)2250  0.0510000  0.0684147   0.745   0.4577  
as.factor(dds$BAN_ID)2251  0.0105521  0.0684147   0.154   0.8777  
as.factor(dds$BAN_ID)2252  0.0923564  0.0684147   1.350   0.1800  
as.factor(dds$BAN_ID)2253 -0.0439728  0.0684147  -0.643   0.5218  
as.factor(dds$BAN_ID)2254 -0.0789100  0.0611919  -1.290   0.2002  
as.factor(dds$BAN_ID)2255 -0.0783825  0.0684147  -1.146   0.2546  
as.factor(dds$BAN_ID)2256  0.0198247  0.0611919   0.324   0.7466  
as.factor(dds$BAN_ID)2257  0.0832525  0.0684147   1.217   0.2265  
as.factor(dds$BAN_ID)2258 -0.0428031  0.0611919  -0.699   0.4859  
as.factor(dds$BAN_ID)2260 -0.0020074  0.0611919  -0.033   0.9739  
as.factor(dds$BAN_ID)2261 -0.0452668  0.0684147  -0.662   0.5097  
as.factor(dds$BAN_ID)2262  0.0229617  0.0684147   0.336   0.7378  
as.factor(dds$BAN_ID)2266 -0.0150167  0.0611919  -0.245   0.8066  
as.factor(dds$BAN_ID)2267 -0.0007241  0.0684147  -0.011   0.9916  
as.factor(dds$BAN_ID)2268 -0.0119999  0.0611919  -0.196   0.8449  
as.factor(dds$BAN_ID)2269  0.0128632  0.0684147   0.188   0.8512  
as.factor(dds$BAN_ID)2270  0.0607913  0.0684147   0.889   0.3763  
as.factor(dds$BAN_ID)2271 -0.0015371  0.0611919  -0.025   0.9800  
as.factor(dds$BAN_ID)2272  0.0319799  0.0611919   0.523   0.6024  
as.factor(dds$BAN_ID)2274 -0.1388462  0.0611919  -2.269   0.0254 *
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.07494 on 101 degrees of freedom
Multiple R-squared:  0.4327,    Adjusted R-squared:  0.1294 
F-statistic: 1.427 on 54 and 101 DF,  p-value: 0.06263</code></pre>
<pre class="r"><code>summary(lm(U[,2]~as.factor(dds$BAN_ID)))</code></pre>
<pre><code>
Call:
lm(formula = U[, 2] ~ as.factor(dds$BAN_ID))

Residuals:
      Min        1Q    Median        3Q       Max 
-0.169707 -0.038096 -0.002767  0.037094  0.163195 

Coefficients:
                            Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept)               -0.0534596  0.0451401  -1.184   0.2391  
as.factor(dds$BAN_ID)2202  0.0591867  0.0570982   1.037   0.3024  
as.factor(dds$BAN_ID)2203  0.0002828  0.0713727   0.004   0.9968  
as.factor(dds$BAN_ID)2204  0.0577802  0.0638377   0.905   0.3676  
as.factor(dds$BAN_ID)2205  0.0737124  0.0713727   1.033   0.3042  
as.factor(dds$BAN_ID)2206  0.0826158  0.0638377   1.294   0.1986  
as.factor(dds$BAN_ID)2207  0.0879071  0.0638377   1.377   0.1715  
as.factor(dds$BAN_ID)2208  0.0213395  0.0638377   0.334   0.7389  
as.factor(dds$BAN_ID)2209  0.0320359  0.0570982   0.561   0.5760  
as.factor(dds$BAN_ID)2210 -0.0136191  0.0638377  -0.213   0.8315  
as.factor(dds$BAN_ID)2212  0.0626723  0.0713727   0.878   0.3820  
as.factor(dds$BAN_ID)2215  0.0311690  0.0638377   0.488   0.6264  
as.factor(dds$BAN_ID)2216  0.0524259  0.0638377   0.821   0.4134  
as.factor(dds$BAN_ID)2218  0.0850026  0.0570982   1.489   0.1397  
as.factor(dds$BAN_ID)2219 -0.0134441  0.0638377  -0.211   0.8336  
as.factor(dds$BAN_ID)2220  0.0730325  0.0570982   1.279   0.2038  
as.factor(dds$BAN_ID)2221 -0.0029708  0.0638377  -0.047   0.9630  
as.factor(dds$BAN_ID)2222  0.0527135  0.0713727   0.739   0.4619  
as.factor(dds$BAN_ID)2224  0.1456428  0.0638377   2.281   0.0246 *
as.factor(dds$BAN_ID)2226  0.0083626  0.0570982   0.146   0.8838  
as.factor(dds$BAN_ID)2228  0.0866913  0.0570982   1.518   0.1321  
as.factor(dds$BAN_ID)2229  0.1241635  0.0713727   1.740   0.0850 .
as.factor(dds$BAN_ID)2232  0.1252020  0.0638377   1.961   0.0526 .
as.factor(dds$BAN_ID)2233 -0.0160478  0.0638377  -0.251   0.8020  
as.factor(dds$BAN_ID)2234  0.1098822  0.0638377   1.721   0.0883 .
as.factor(dds$BAN_ID)2235  0.0877375  0.0638377   1.374   0.1724  
as.factor(dds$BAN_ID)2236  0.0278832  0.0713727   0.391   0.6969  
as.factor(dds$BAN_ID)2239  0.0023093  0.0638377   0.036   0.9712  
as.factor(dds$BAN_ID)2240  0.0688619  0.0713727   0.965   0.3369  
as.factor(dds$BAN_ID)2242  0.0677578  0.0638377   1.061   0.2910  
as.factor(dds$BAN_ID)2243  0.0887655  0.0638377   1.390   0.1674  
as.factor(dds$BAN_ID)2245  0.1602406  0.0713727   2.245   0.0269 *
as.factor(dds$BAN_ID)2247  0.0476049  0.0713727   0.667   0.5063  
as.factor(dds$BAN_ID)2248  0.1460913  0.0638377   2.288   0.0242 *
as.factor(dds$BAN_ID)2249 -0.0001295  0.0713727  -0.002   0.9986  
as.factor(dds$BAN_ID)2250  0.0665557  0.0713727   0.933   0.3533  
as.factor(dds$BAN_ID)2251  0.1058272  0.0713727   1.483   0.1413  
as.factor(dds$BAN_ID)2252  0.0573114  0.0713727   0.803   0.4239  
as.factor(dds$BAN_ID)2253  0.0765794  0.0713727   1.073   0.2859  
as.factor(dds$BAN_ID)2254  0.0286648  0.0638377   0.449   0.6544  
as.factor(dds$BAN_ID)2255  0.0730069  0.0713727   1.023   0.3088  
as.factor(dds$BAN_ID)2256  0.0260126  0.0638377   0.407   0.6845  
as.factor(dds$BAN_ID)2257  0.0862133  0.0713727   1.208   0.2299  
as.factor(dds$BAN_ID)2258 -0.0379588  0.0638377  -0.595   0.5534  
as.factor(dds$BAN_ID)2260  0.0679122  0.0638377   1.064   0.2899  
as.factor(dds$BAN_ID)2261  0.0269032  0.0713727   0.377   0.7070  
as.factor(dds$BAN_ID)2262  0.0588332  0.0713727   0.824   0.4117  
as.factor(dds$BAN_ID)2266  0.0662375  0.0638377   1.038   0.3019  
as.factor(dds$BAN_ID)2267 -0.0026719  0.0713727  -0.037   0.9702  
as.factor(dds$BAN_ID)2268  0.0820324  0.0638377   1.285   0.2017  
as.factor(dds$BAN_ID)2269 -0.0838104  0.0713727  -1.174   0.2431  
as.factor(dds$BAN_ID)2270 -0.0509226  0.0713727  -0.713   0.4772  
as.factor(dds$BAN_ID)2271  0.0608959  0.0638377   0.954   0.3424  
as.factor(dds$BAN_ID)2272  0.1565321  0.0638377   2.452   0.0159 *
as.factor(dds$BAN_ID)2274  0.0072852  0.0638377   0.114   0.9094  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.07818 on 101 degrees of freedom
Multiple R-squared:  0.3826,    Adjusted R-squared:  0.0525 
F-statistic: 1.159 on 54 and 101 DF,  p-value: 0.2593</code></pre>
<pre class="r"><code>summary(lm(U[,1]~as.factor(dds$Time)))</code></pre>
<pre><code>
Call:
lm(formula = U[, 1] ~ as.factor(dds$Time))

Residuals:
      Min        1Q    Median        3Q       Max 
-0.239457 -0.037495 -0.002248  0.045992  0.187738 

Coefficients:
                     Estimate Std. Error t value Pr(&gt;|t|)   
(Intercept)          -0.02135    0.01055  -2.024   0.0448 * 
as.factor(dds$Time)2  0.03139    0.01492   2.104   0.0370 * 
as.factor(dds$Time)3  0.01853    0.01707   1.086   0.2794   
as.factor(dds$Time)4  0.06858    0.03364   2.039   0.0432 * 
as.factor(dds$Time)5  0.09375    0.03364   2.787   0.0060 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.07824 on 151 degrees of freedom
Multiple R-squared:  0.07573,   Adjusted R-squared:  0.05124 
F-statistic: 3.093 on 4 and 151 DF,  p-value: 0.01759</code></pre>
<pre class="r"><code>summary(lm(U[,2]~as.factor(dds$Time)))</code></pre>
<pre><code>
Call:
lm(formula = U[, 2] ~ as.factor(dds$Time))

Residuals:
      Min        1Q    Median        3Q       Max 
-0.200093 -0.047013 -0.008264  0.045612  0.197906 

Coefficients:
                      Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)          -0.039276   0.009872  -3.978 0.000107 ***
as.factor(dds$Time)2  0.079112   0.013961   5.666 7.17e-08 ***
as.factor(dds$Time)3  0.026810   0.015972   1.678 0.095320 .  
as.factor(dds$Time)4  0.066983   0.031478   2.128 0.034967 *  
as.factor(dds$Time)5  0.077072   0.031478   2.448 0.015492 *  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.07321 on 151 degrees of freedom
Multiple R-squared:  0.1906,    Adjusted R-squared:  0.1691 
F-statistic: 8.889 on 4 and 151 DF,  p-value: 1.796e-06</code></pre>
<pre class="r"><code># Variance explained 

varex = 0
cumvar = 0
denom = sum(D^2)
for(i in 1:64){
  varex[i] = D[i]^2/denom
  cumvar[i] = sum(D[1:i]^2)/denom
}

## variance explained by each PC cumulatively
varex</code></pre>
<pre><code> [1] 0.137853004 0.076769100 0.060778451 0.053842661 0.040389428
 [6] 0.023186916 0.018280595 0.017486529 0.016129700 0.015261370
[11] 0.014354599 0.013283976 0.012225001 0.011367097 0.010638096
[16] 0.009655865 0.009472176 0.009013108 0.008693300 0.008556714
[21] 0.008241348 0.007990630 0.007529435 0.007167652 0.006986356
[26] 0.006751876 0.006646510 0.006307294 0.006207828 0.006077119
[31] 0.005940811 0.005649890 0.005522526 0.005381654 0.005346253
[36] 0.005315039 0.005203941 0.005054253 0.004977886 0.004926546
[41] 0.004900597 0.004734438 0.004698370 0.004660227 0.004559344
[46] 0.004484403 0.004458207 0.004392114 0.004296109 0.004198661
[51] 0.004186982 0.004173545 0.004142836 0.004036708 0.004019439
[56] 0.004002805 0.003955744 0.003912123 0.003854044 0.003821618
[61] 0.003748342 0.003737302 0.003672817 0.003624890</code></pre>
<pre class="r"><code>cumvar</code></pre>
<pre><code> [1] 0.1378530 0.2146221 0.2754006 0.3292432 0.3696326 0.3928196 0.4111002
 [8] 0.4285867 0.4447164 0.4599778 0.4743324 0.4876163 0.4998413 0.5112084
[15] 0.5218465 0.5315024 0.5409746 0.5499877 0.5586810 0.5672377 0.5754790
[22] 0.5834697 0.5909991 0.5981667 0.6051531 0.6119050 0.6185515 0.6248588
[29] 0.6310666 0.6371437 0.6430845 0.6487344 0.6542570 0.6596386 0.6649849
[36] 0.6702999 0.6755038 0.6805581 0.6855360 0.6904625 0.6953631 0.7000976
[43] 0.7047959 0.7094562 0.7140155 0.7184999 0.7229581 0.7273502 0.7316463
[50] 0.7358450 0.7400320 0.7442055 0.7483484 0.7523851 0.7564045 0.7604073
[57] 0.7643631 0.7682752 0.7721292 0.7759508 0.7796992 0.7834365 0.7871093
[64] 0.7907342</code></pre>
<pre class="r"><code>U &lt;- as.data.frame(U)
norm_count &lt;- ggplot(data=U, aes(x=U[,3], y=U[,4], color=factor(dds$Time))) + geom_point() + scale_color_discrete(name = &quot;Time&quot;) + xlab(&quot;PC3 (6% of variation explained)&quot;) + ylab(&quot;PC4 (5% of variation explained)&quot;) 

plot_grid(norm_count)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-7-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>U &lt;- as.data.frame(U)
norm_count &lt;- ggplot(data=U, aes(x=U[,5], y=U[,6], color=factor(dds$Time))) + geom_point() + scale_color_discrete(name = &quot;Time&quot;) + xlab(&quot;PC5 (4% of variation explained)&quot;) + ylab(&quot;PC6 (5% of variation explained)&quot;) 

plot_grid(norm_count)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-7-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>U &lt;- as.data.frame(U)
norm_count &lt;- ggplot(data=U, aes(x=U[,7], y=U[,8], color=factor(dds$Time))) + geom_point() + scale_color_discrete(name = &quot;Time&quot;) + xlab(&quot;PC7&quot;) + ylab(&quot;PC6&quot;) 

plot_grid(norm_count)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-7-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Flow cell

technical_sample_info &lt;- read.csv(&quot;~/Desktop/weight_restoration/weight_restoration/data/technical_sample_info.csv&quot;)

U &lt;- as.data.frame(U)
norm_count &lt;- ggplot(data=U, aes(x=U[,1], y=U[,2], color=factor(technical_sample_info$Flowell))) + geom_point() + scale_color_discrete(name = &quot;Flowcell&quot;) + xlab(&quot;PC1 (14% of variation explained)&quot;) + ylab(&quot;PC2 (8% of variation explained)&quot;)

plot_grid(norm_count)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-7-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save_plot(&quot;/Users/laurenblake/Dropbox/Figures/PCA_flowcell.png&quot;, norm_count,
          base_aspect_ratio = 1, base_width = 6.95, base_height = 5.25)</code></pre>
</div>
<div id="differential-expression-analysis-with-voomlimma" class="section level1">
<h1>Differential expression analysis with voom+limma</h1>
<pre class="r"><code>library(edgeR)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre><code>
Attaching package: &#39;limma&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:DESeq2&#39;:

    plotMA</code></pre>
<pre><code>The following object is masked from &#39;package:BiocGenerics&#39;:

    plotMA</code></pre>
<pre class="r"><code>library(limma)


tx.salmon.limma &lt;- tximport(salmon.files, type = &quot;salmon&quot;, tx2gene = tx2gene, dropInfReps=TRUE, countsFromAbundance = &quot;lengthScaledTPM&quot;)</code></pre>
<pre><code>reading in files with read_tsv</code></pre>
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 
summarizing abundance
summarizing counts
summarizing length</code></pre>
<pre class="r"><code># Basic filtering
cpm_filtered &lt;- (rowSums(tx.salmon.limma$counts &gt; 2) &gt; 117)

genes_in_cutoff &lt;- tx.salmon.limma$counts[cpm_filtered==TRUE,]

y &lt;- DGEList(genes_in_cutoff)
y &lt;- calcNormFactors(y)

design &lt;- model.matrix(~as.factor(Time), data = clinical_sample)
colnames(design) &lt;- c(&quot;Intercept&quot;, &quot;Time2&quot;, &quot;Time3&quot;, &quot;Time4&quot;, &quot;Time5&quot;)


cpm.voom &lt;- voom(y, design, normalize.method=&quot;none&quot;)
#check_rel &lt;- duplicateCorrelation(cpm.voom, design, block = clinical_sample$BAN_ID)
check_rel_correlation &lt;- 0.09083534
cpm.voom.corfit &lt;- voom(y, design, normalize.method=&quot;none&quot;, plot = TRUE, block = clinical_sample$BAN_ID, correlation = check_rel_correlation)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotDensities(cpm.voom.corfit[,3])</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_genes &lt;- prcomp(t(cpm.voom.corfit$E), scale = T, retx = TRUE, center = TRUE)

matrixpca &lt;- pca_genes$x
PC1 &lt;- matrixpca[,1]
PC2 &lt;- matrixpca[,2]
pc3 &lt;- matrixpca[,3]
pc4 &lt;- matrixpca[,4]
pc5 &lt;- matrixpca[,5]

pcs &lt;- data.frame(PC1, PC2, pc3, pc4, pc5)

summary &lt;- summary(pca_genes)

head(summary$importance[2,1:5])</code></pre>
<pre><code>    PC1     PC2     PC3     PC4     PC5 
0.16005 0.08588 0.07426 0.05609 0.02964 </code></pre>
<pre class="r"><code>pca_norm_count &lt;- ggplot(data=pcs, aes(x=PC1, y=PC2, color=clinical_sample$Time)) + geom_point(aes(colour = as.factor(clinical_sample$Time))) + ggtitle(&quot;PCA of normalized counts&quot;) + scale_color_discrete(name = &quot;Time&quot;)

plot_grid(pca_norm_count)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run PCA

X = t(scale(t(assay(vsd)),center=TRUE,scale=FALSE))
sv = svd(t(X))
U = sv$u
V = sv$v
D = sv$d
## in R calculate the rank of a matrix is by
#qr(t(X))$rank

plot(U[,1],U[,2],xlab=&quot;PC1&quot;,ylab=&quot;PC2&quot;)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-8-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>U &lt;- as.data.frame(U)
norm_count &lt;- ggplot(data=U, aes(x=-U[,1], y=-U[,2], color=clinical_sample[,2])) + geom_point(aes(colour = as.factor(clinical_sample[,2]))) + ggtitle(&quot;PCA of normalized counts&quot;) + scale_color_discrete(name = &quot;Time&quot;) + xlab(&quot;PC1&quot;) + ylab(&quot;PC2&quot;)

plot_grid(norm_count)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-8-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Variance explained 

varex = 0
cumvar = 0
denom = sum(D^2)
for(i in 1:64){
  varex[i] = D[i]^2/denom
  cumvar[i] = sum(D[1:i]^2)/denom
}

## variance explained by each PC cumulatively
varex</code></pre>
<pre><code> [1] 0.137853004 0.076769100 0.060778451 0.053842661 0.040389428
 [6] 0.023186916 0.018280595 0.017486529 0.016129700 0.015261370
[11] 0.014354599 0.013283976 0.012225001 0.011367097 0.010638096
[16] 0.009655865 0.009472176 0.009013108 0.008693300 0.008556714
[21] 0.008241348 0.007990630 0.007529435 0.007167652 0.006986356
[26] 0.006751876 0.006646510 0.006307294 0.006207828 0.006077119
[31] 0.005940811 0.005649890 0.005522526 0.005381654 0.005346253
[36] 0.005315039 0.005203941 0.005054253 0.004977886 0.004926546
[41] 0.004900597 0.004734438 0.004698370 0.004660227 0.004559344
[46] 0.004484403 0.004458207 0.004392114 0.004296109 0.004198661
[51] 0.004186982 0.004173545 0.004142836 0.004036708 0.004019439
[56] 0.004002805 0.003955744 0.003912123 0.003854044 0.003821618
[61] 0.003748342 0.003737302 0.003672817 0.003624890</code></pre>
<pre class="r"><code>cumvar</code></pre>
<pre><code> [1] 0.1378530 0.2146221 0.2754006 0.3292432 0.3696326 0.3928196 0.4111002
 [8] 0.4285867 0.4447164 0.4599778 0.4743324 0.4876163 0.4998413 0.5112084
[15] 0.5218465 0.5315024 0.5409746 0.5499877 0.5586810 0.5672377 0.5754790
[22] 0.5834697 0.5909991 0.5981667 0.6051531 0.6119050 0.6185515 0.6248588
[29] 0.6310666 0.6371437 0.6430845 0.6487344 0.6542570 0.6596386 0.6649849
[36] 0.6702999 0.6755038 0.6805581 0.6855360 0.6904625 0.6953631 0.7000976
[43] 0.7047959 0.7094562 0.7140155 0.7184999 0.7229581 0.7273502 0.7316463
[50] 0.7358450 0.7400320 0.7442055 0.7483484 0.7523851 0.7564045 0.7604073
[57] 0.7643631 0.7682752 0.7721292 0.7759508 0.7796992 0.7834365 0.7871093
[64] 0.7907342</code></pre>
<pre class="r"><code># Run lmFit and eBayes in limma

fit &lt;- lmFit(cpm.voom.corfit, design, block=clinical_sample$BAN_ID, correlation=check_rel_correlation)

# In the contrast matrix, have the time points

cm1 &lt;- makeContrasts(Time1v2 = Time2, Time2v3 = Time3 - Time2, levels = design)

# Fit the new model
diff_species &lt;- contrasts.fit(fit, cm1)
fit1 &lt;- eBayes(diff_species)

FDR_level &lt;- 0.01

Time1v2 =topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
Time2v3 =topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)

dim(Time1v2[which(Time1v2$adj.P.Val &lt; FDR_level),])</code></pre>
<pre><code>[1] 759   6</code></pre>
<pre class="r"><code>dim(Time1v2[which(Time2v3$adj.P.Val &lt; FDR_level),])</code></pre>
<pre><code>[1] 986   6</code></pre>
</div>
<div id="differential-expression-analysis-tmm-limmavoom" class="section level1">
<h1>Differential expression analysis TMM, limma+voom</h1>
<pre class="r"><code>log_counts_genes &lt;- as.data.frame(log2(tx.salmon$counts))
cpm &lt;- cpm(tx.salmon$counts, log=TRUE)

expr_cutoff &lt;- 1.5
hist(cpm, main = &quot;log2(CPM) values in unfiltered data&quot;, breaks = 100, xlab = &quot;log2(CPM) values&quot;)
abline(v = expr_cutoff, col = &quot;red&quot;, lwd = 3)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Basic filtering
cpm_filtered &lt;- (rowSums(cpm &gt; 1.5) &gt; 76)

genes_in_cutoff &lt;- cpm[cpm_filtered==TRUE,]

hist(as.numeric(unlist(genes_in_cutoff)), main = &quot;log2(CPM) values in filtered data (n = 156 samples, 12,152 genes)&quot;, breaks = 100, xlab = &quot;log2(CPM) values&quot;)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Find the original counts of all of the genes that fit the criteria 

counts_genes_in_cutoff &lt;- tx.salmon$counts[cpm_filtered==TRUE,]
dim(counts_genes_in_cutoff)</code></pre>
<pre><code>[1] 11438   156</code></pre>
<pre class="r"><code># Take the TMM of the counts only for the genes that remain after filtering

dge_in_cutoff &lt;- DGEList(counts=as.matrix(counts_genes_in_cutoff), genes=rownames(counts_genes_in_cutoff), group = as.character(t(clinical_sample$BAN_ID)))
dge_in_cutoff &lt;- calcNormFactors(dge_in_cutoff)

cpm_in_cutoff &lt;- cpm(dge_in_cutoff, normalized.lib.sizes=TRUE, log=TRUE)

pca_genes &lt;- prcomp(t(cpm_in_cutoff), scale = T, retx = TRUE, center = TRUE)

matrixpca &lt;- pca_genes$x
PC1 &lt;- matrixpca[,1]
PC2 &lt;- matrixpca[,2]
pc3 &lt;- matrixpca[,3]
pc4 &lt;- matrixpca[,4]
pc5 &lt;- matrixpca[,5]

pcs &lt;- data.frame(PC1, PC2, pc3, pc4, pc5)

summary &lt;- summary(pca_genes)

head(summary$importance[2,1:5])</code></pre>
<pre><code>    PC1     PC2     PC3     PC4     PC5 
0.23534 0.13424 0.10007 0.04892 0.03726 </code></pre>
<pre class="r"><code>norm_count &lt;- ggplot(data=pcs, aes(x=PC1, y=PC2, color= as.factor(clinical_sample$BAN_ID))) + geom_point(aes(colour = as.factor(clinical_sample$BAN_ID))) + ggtitle(&quot;PCA of normalized counts&quot;) + scale_color_discrete(name = &quot;Time&quot;)

plot_grid(norm_count)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>design &lt;- model.matrix(~as.factor(Time), data = clinical_sample)
colnames(design) &lt;- c(&quot;Intercept&quot;, &quot;Time2&quot;, &quot;Time3&quot;, &quot;Time4&quot;, &quot;Time5&quot;)


cpm.voom &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;none&quot;)
#check_rel &lt;- duplicateCorrelation(cpm.voom, design, block = clinical_sample$BAN_ID)
check_rel_correlation &lt;-  0.1136422
cpm.voom.corfit &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;none&quot;, plot = TRUE, block = clinical_sample$BAN_ID, correlation = check_rel_correlation)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-9-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotDensities(cpm.voom.corfit[,1])</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-9-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotDensities(cpm.voom.corfit[,2])</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-9-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotDensities(cpm.voom.corfit[,3])</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-9-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_genes &lt;- prcomp(t(cpm.voom.corfit$E), scale = T, retx = TRUE, center = TRUE)

matrixpca &lt;- pca_genes$x
PC1 &lt;- matrixpca[,1]
PC2 &lt;- matrixpca[,2]
pc3 &lt;- matrixpca[,3]
pc4 &lt;- matrixpca[,4]
pc5 &lt;- matrixpca[,5]

pcs &lt;- data.frame(PC1, PC2, pc3, pc4, pc5)

summary &lt;- summary(pca_genes)

head(summary$importance[2,1:5])</code></pre>
<pre><code>    PC1     PC2     PC3     PC4     PC5 
0.23582 0.13441 0.10035 0.04874 0.03730 </code></pre>
<pre class="r"><code>ggplot(data=pcs, aes(x=PC1, y=PC2, color=clinical_sample$Time)) + geom_point(aes(colour = as.factor(clinical_sample$Time))) + ggtitle(&quot;PCA of normalized counts&quot;) + scale_color_discrete(name = &quot;Time&quot;)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-9-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run PCA

X = t(scale(t(cpm_in_cutoff),center=TRUE,scale=FALSE))
sv = svd(t(X))
U = sv$u
V = sv$v
D = sv$d
## in R calculate the rank of a matrix is by
#qr(t(X))$rank

plot(U[,1],U[,2],xlab=&quot;PC1&quot;,ylab=&quot;PC2&quot;)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-9-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>U &lt;- as.data.frame(U)
norm_count &lt;- ggplot(data=U, aes(x=U[,1], y=U[,2], color=clinical_sample[,2])) + geom_point(aes(colour = as.factor(clinical_sample[,2]))) + ggtitle(&quot;PCA of normalized counts&quot;) + scale_color_discrete(name = &quot;Time&quot;) + xlab(&quot;PC1&quot;) + ylab(&quot;PC2&quot;)

plot_grid(norm_count)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-9-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Variance explained 

varex = 0
cumvar = 0
denom = sum(D^2)
for(i in 1:64){
  varex[i] = D[i]^2/denom
  cumvar[i] = sum(D[1:i]^2)/denom
}

## variance explained by each PC cumulatively
varex</code></pre>
<pre><code> [1] 0.185931394 0.095561275 0.073591851 0.053321414 0.046532587
 [6] 0.030354366 0.026293663 0.024792774 0.020620619 0.019672108
[11] 0.017365141 0.016098097 0.015036299 0.013182331 0.012626453
[16] 0.011835102 0.010965464 0.010458941 0.009875649 0.009349001
[21] 0.009057611 0.008331003 0.007813140 0.007670993 0.007334160
[26] 0.006914086 0.006625423 0.006463381 0.006073664 0.005858615
[31] 0.005811333 0.005618453 0.005372810 0.005172660 0.004803311
[36] 0.004583213 0.004526531 0.004430815 0.004297253 0.004157526
[41] 0.004032902 0.003968551 0.003926069 0.003840625 0.003734090
[46] 0.003677950 0.003626398 0.003533522 0.003354889 0.003325657
[51] 0.003218755 0.003181576 0.003129086 0.003075036 0.002940664
[56] 0.002850641 0.002812216 0.002758044 0.002696029 0.002675502
[61] 0.002608802 0.002544542 0.002485642 0.002389173</code></pre>
<pre class="r"><code>cumvar</code></pre>
<pre><code> [1] 0.1859314 0.2814927 0.3550845 0.4084059 0.4549385 0.4852929 0.5115866
 [8] 0.5363793 0.5569999 0.5766721 0.5940372 0.6101353 0.6251716 0.6383539
[15] 0.6509804 0.6628155 0.6737809 0.6842399 0.6941155 0.7034645 0.7125221
[22] 0.7208531 0.7286663 0.7363373 0.7436714 0.7505855 0.7572109 0.7636743
[29] 0.7697480 0.7756066 0.7814179 0.7870364 0.7924092 0.7975819 0.8023852
[36] 0.8069684 0.8114949 0.8159257 0.8202230 0.8243805 0.8284134 0.8323820
[43] 0.8363080 0.8401487 0.8438827 0.8475607 0.8511871 0.8547206 0.8580755
[50] 0.8614012 0.8646199 0.8678015 0.8709306 0.8740056 0.8769463 0.8797969
[57] 0.8826091 0.8853672 0.8880632 0.8907387 0.8933475 0.8958921 0.8983777
[64] 0.9007669</code></pre>
<pre class="r"><code># Reversed

U &lt;- as.data.frame(U)
norm_count &lt;- ggplot(data=U, aes(x=-U[,1], y=-U[,2], color=clinical_sample[,2])) + geom_point(aes(colour = as.factor(clinical_sample[,2]))) + ggtitle(&quot;PCA of normalized counts (rotated)&quot;) + scale_color_discrete(name = &quot;Time&quot;) + xlab(&quot;PC1&quot;) + ylab(&quot;PC2&quot;)

plot_grid(norm_count)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-9-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run lmFit and eBayes in limma

fit &lt;- lmFit(cpm.voom.corfit, design, block=clinical_sample$BAN_ID, correlation=check_rel_correlation)

# In the contrast matrix, have the time points

cm1 &lt;- makeContrasts(Time1v2 = Time2, Time2v3 = Time3 - Time2, levels = design)

# Fit the new model
diff_species &lt;- contrasts.fit(fit, cm1)
fit1 &lt;- eBayes(diff_species)

FDR_level &lt;- 0.01

Time1v2 =topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
Time2v3 =topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)

dim(Time1v2[which(Time1v2$adj.P.Val &lt; FDR_level),])</code></pre>
<pre><code>[1] 1036    7</code></pre>
<pre class="r"><code>dim(Time2v3[which(Time2v3$adj.P.Val &lt; FDR_level),])</code></pre>
<pre><code>[1] 994   7</code></pre>
<pre class="r"><code>topTable(fit1, coef=1, adjust=&quot;BH&quot;, number=100, sort.by=&quot;T&quot;)</code></pre>
<pre><code>                    genes     logFC   AveExpr        t      P.Value
DCAF6               DCAF6 0.6695334  5.156446 7.842206 6.585794e-13
CHPT1               CHPT1 1.0720311  5.004514 7.436520 6.506927e-12
RNF10               RNF10 1.1048642  8.504369 7.421656 7.069507e-12
TSPAN5             TSPAN5 1.1493148  4.878885 7.216883 2.198913e-11
MICAL2             MICAL2 0.8325314  6.949515 7.183476 2.642432e-11
ALAS2               ALAS2 2.0197411  9.854164 7.165553 2.915723e-11
XK                     XK 1.5183572  3.239358 7.140213 3.350323e-11
IFIT1B             IFIT1B 1.5843831  3.701225 7.032988 6.015869e-11
LINC00570       LINC00570 1.5720373  2.519723 7.029168 6.142158e-11
DCAF12             DCAF12 1.2878999  8.231272 7.023604 6.330818e-11
GID4                 GID4 0.6285829  3.697393 7.014620 6.647552e-11
AC009469.1     AC009469.1 1.7741990  2.891726 6.990244 7.587920e-11
PPM1A               PPM1A 0.8024280  6.260500 6.931849 1.040829e-10
IGF2BP2           IGF2BP2 1.2514240  5.101798 6.909931 1.171530e-10
SLC6A8             SLC6A8 1.4654227  5.821905 6.902783 1.217560e-10
MKRN1               MKRN1 1.0072377  8.198901 6.874338 1.419078e-10
ADIPOR1           ADIPOR1 1.1578874  9.715771 6.831520 1.785981e-10
SPTB                 SPTB 1.1014652  5.381431 6.814813 1.953263e-10
MPP1                 MPP1 0.9014442  7.519320 6.814091 1.960830e-10
TMEM63B           TMEM63B 0.8075932  5.463581 6.778473 2.372308e-10
STRADB             STRADB 1.4497870  7.154012 6.758343 2.641378e-10
LCN2                 LCN2 1.5986821  3.974926 6.747749 2.794852e-10
HAGH                 HAGH 1.0741095  6.450115 6.728931 3.089434e-10
SLC4A1             SLC4A1 1.7216598  8.336649 6.728668 3.093775e-10
MARK3               MARK3 0.4726809  6.408598 6.727919 3.106128e-10
MCOLN1             MCOLN1 0.7714675  5.873899 6.724095 3.169968e-10
SESN3               SESN3 0.7924189  5.961494 6.716698 3.297142e-10
SELENBP1         SELENBP1 1.8118487  6.485878 6.703110 3.544013e-10
VTI1B               VTI1B 0.5360321  5.918813 6.700097 3.601167e-10
OR2W3               OR2W3 1.5390544  5.459522 6.696823 3.664315e-10
PSME4               PSME4 0.4474220  5.509253 6.695308 3.693897e-10
AGO2                 AGO2 0.7583776  4.594308 6.695283 3.694402e-10
MXI1                 MXI1 1.1567357  6.213960 6.684992 3.901732e-10
CCDC176           CCDC176 1.1335950  3.033738 6.671380 4.193654e-10
TRIM10             TRIM10 1.3977127  2.254365 6.664246 4.355143e-10
TMOD1               TMOD1 1.4083392  4.757680 6.657436 4.515022e-10
MBNL3               MBNL3 1.0546141  6.591863 6.655955 4.550553e-10
SNCA                 SNCA 1.5554085  7.039598 6.620823 5.478908e-10
ARHGEF12         ARHGEF12 0.9842513  4.505014 6.614403 5.667664e-10
TRIM58             TRIM58 1.4917264  5.591949 6.610865 5.774393e-10
YBX3                 YBX3 1.3187127  8.731469 6.606821 5.898830e-10
GMPR                 GMPR 1.4192950  5.901969 6.598837 6.152316e-10
MAP2K3             MAP2K3 0.7252218  7.574698 6.584250 6.643441e-10
GPR146             GPR146 1.3326144  4.417140 6.548413 8.020010e-10
BPGM                 BPGM 1.3953525  5.318873 6.527727 8.938693e-10
TNS1                 TNS1 1.4499061  6.502811 6.507205 9.952230e-10
C7orf73           C7orf73 0.7904275  5.524864 6.502527 1.019866e-09
FECH                 FECH 1.3065676  5.170922 6.494579 1.063118e-09
FAM104A           FAM104A 0.8352853  6.326572 6.463191 1.252286e-09
RNF123             RNF123 0.7136580  5.985845 6.455809 1.301381e-09
KIAA0430         KIAA0430 0.4491244  7.202189 6.445302 1.374538e-09
GLRX5               GLRX5 1.2447421  6.076871 6.436463 1.439196e-09
CAMP                 CAMP 1.3912933  3.124174 6.417481 1.588355e-09
JAZF1               JAZF1 0.7888037  5.853919 6.415630 1.603693e-09
PINK1               PINK1 0.7705890  7.051630 6.415365 1.605894e-09
PPP2R5B           PPP2R5B 0.6506912  4.784111 6.410804 1.644354e-09
BPI                   BPI 1.2759594  3.417495 6.404795 1.696416e-09
NPRL3               NPRL3 0.9520682  7.839812 6.400940 1.730659e-09
FURIN               FURIN 0.5607257  7.783370 6.398254 1.754916e-09
EPB42               EPB42 1.4548888  6.378389 6.391533 1.817103e-09
MICALCL           MICALCL 0.9768268  1.960105 6.381087 1.918081e-09
BCL2L1             BCL2L1 1.3172600  8.903068 6.342768 2.338017e-09
RANBP10           RANBP10 0.7453902  6.018886 6.330192 2.494615e-09
AP2A1               AP2A1 0.5033551  7.479892 6.319089 2.641424e-09
MARCH8             MARCH8 0.9837886  7.715584 6.300426 2.907538e-09
FAM210B           FAM210B 1.2831587  7.607063 6.284853 3.149627e-09
KEL                   KEL 1.2609124  2.514250 6.269176 3.413319e-09
ZER1                 ZER1 0.6864264  7.095405 6.268258 3.429418e-09
RPIA                 RPIA 0.6932537  4.349177 6.266644 3.457900e-09
LTF                   LTF 1.6714831  3.944821 6.262062 3.540036e-09
FBXO7               FBXO7 1.0609317  9.439777 6.238444 3.994749e-09
NCOA4               NCOA4 0.6123100  8.798708 6.231920 4.130155e-09
FBXO9               FBXO9 0.6665178  5.959144 6.229167 4.188651e-09
NFIX                 NFIX 1.1428774  5.827849 6.224884 4.281246e-09
FGFR1OP2         FGFR1OP2 0.6222712  4.942062 6.219585 4.398622e-09
EIF1B               EIF1B 0.5851047  5.591378 6.215836 4.483575e-09
ANK1                 ANK1 1.2147838  5.459370 6.203030 4.786036e-09
EIF2AK1           EIF2AK1 0.7326650  7.481181 6.187417 5.182042e-09
LGALS3             LGALS3 0.9802526  6.961567 6.181336 5.344817e-09
GSPT1               GSPT1 0.9928710  7.746085 6.173114 5.572928e-09
GATA1               GATA1 1.1172246  5.129318 6.172819 5.581298e-09
CTNNAL1           CTNNAL1 1.2732015  1.660801 6.143089 6.490133e-09
PBX1                 PBX1 1.0481146  2.699980 6.132908 6.833580e-09
SLC25A37         SLC25A37 0.8925935 10.934617 6.132857 6.835337e-09
CREG1               CREG1 0.6659727  5.771756 6.131528 6.881491e-09
FAM214B           FAM214B 0.5452537  6.609295 6.127967 7.006630e-09
TPGS2               TPGS2 0.8450435  5.939135 6.110924 7.637234e-09
RILP                 RILP 1.0132267  5.374708 6.096869 8.198914e-09
MYL4                 MYL4 1.2948633  5.505320 6.093069 8.357618e-09
ZRANB1             ZRANB1 0.4085094  5.105642 6.083573 8.767482e-09
MRC2                 MRC2 1.2377456  3.819584 6.082894 8.797527e-09
PIP4K2A           PIP4K2A 0.5459935  7.430127 6.073779 9.210882e-09
ASCC2               ASCC2 0.8757655  7.976084 6.073068 9.243904e-09
ABCC13             ABCC13 1.7737264  2.057536 6.069369 9.417651e-09
OPTN                 OPTN 0.8180892  7.033724 6.065888 9.584055e-09
DMTN                 DMTN 1.2455146  9.433939 6.064216 9.664998e-09
SELK                 SELK 0.6218770  4.167201 6.050717 1.034365e-08
CTD-2336O2.1 CTD-2336O2.1 0.8331313  4.006798 6.040859 1.086859e-08
SLC14A1           SLC14A1 1.1536028  3.791299 6.024706 1.178582e-08
HEMGN               HEMGN 1.1990770  4.559246 6.014145 1.242608e-08
                adj.P.Val         B
DCAF6        7.532832e-09 18.895018
CHPT1        2.695367e-08 16.693017
RNF10        2.695367e-08 16.568541
TSPAN5       5.474428e-08 15.524496
MICAL2       5.474428e-08 15.347705
ALAS2        5.474428e-08 15.163898
XK           5.474428e-08 15.034330
IFIT1B       6.912246e-08 14.523179
LINC00570    6.912246e-08 14.280186
DCAF12       6.912246e-08 14.489214
GID4         6.912246e-08 14.433211
AC009469.1   7.232553e-08 14.205658
PPM1A        9.157692e-08 14.038147
IGF2BP2      9.284297e-08 13.923715
SLC6A8       9.284297e-08 13.887738
MKRN1        1.014463e-07 13.723019
ADIPOR1      1.180420e-07 13.462978
SPTB         1.180420e-07 13.434797
MPP1         1.180420e-07 13.425132
TMEM63B      1.320518e-07 13.248637
STRADB       1.320518e-07 13.142878
LCN2         1.320518e-07 13.079018
HAGH         1.320518e-07 12.995538
SLC4A1       1.320518e-07 12.977523
MARK3        1.320518e-07 12.990418
MCOLN1       1.320518e-07 12.970863
SESN3        1.320518e-07 12.933201
SELENBP1     1.320518e-07 12.863871
VTI1B        1.320518e-07 12.848637
OR2W3        1.320518e-07 12.832156
PSME4        1.320518e-07 12.824316
AGO2         1.320518e-07 12.822934
MXI1         1.352364e-07 12.772039
CCDC176      1.406736e-07 12.624549
TRIM10       1.406736e-07 12.424209
TMOD1        1.406736e-07 12.631782
MBNL3        1.406736e-07 12.624774
SNCA         1.645630e-07 12.446408
ARHGEF12     1.645630e-07 12.414128
TRIM58       1.645630e-07 12.396712
YBX3         1.645630e-07 12.360661
GMPR         1.675481e-07 12.335960
MAP2K3       1.767155e-07 12.260971
GPR146       2.084838e-07 12.081368
BPGM         2.272017e-07 11.978670
TNS1         2.474644e-07 11.876427
C7orf73      2.481964e-07 11.851909
FECH         2.533321e-07 11.812823
FAM104A      2.923194e-07 11.655988
RNF123       2.977039e-07 11.618415
KIAA0430     3.082739e-07 11.568447
GLRX5        3.165678e-07 11.522624
CAMP         3.339676e-07 11.384980
JAZF1        3.339676e-07 11.418438
PINK1        3.339676e-07 11.419810
PPP2R5B      3.358594e-07 11.397158
BPI          3.402158e-07 11.353080
NPRL3        3.402158e-07 11.347523
FURIN        3.402158e-07 11.334798
EPB42        3.464004e-07 11.300516
MICALCL      3.596558e-07 11.010489
BCL2L1       4.313264e-07 11.051500
RANBP10      4.529112e-07 10.995982
AP2A1        4.720720e-07 10.945711
MARCH8       5.116373e-07 10.854109
FAM210B      5.458399e-07 10.778181
KEL          5.732096e-07 10.579008
ZER1         5.732096e-07 10.695990
RPIA         5.732096e-07 10.689174
LTF          5.784420e-07 10.666488
FBXO7        6.435484e-07 10.537501
NCOA4        6.561210e-07 10.515533
FBXO9        6.562985e-07 10.500209
NFIX         6.617418e-07 10.479762
FGFR1OP2     6.708193e-07 10.456842
EIF1B        6.747780e-07 10.435182
ANK1         7.109439e-07 10.373649
EIF2AK1      7.598999e-07 10.303974
LGALS3       7.738483e-07 10.272280
GSPT1        7.881344e-07 10.235417
GATA1        7.881344e-07 10.228879
CTNNAL1      9.052944e-07  9.735502
PBX1         9.260058e-07  9.975571
SLC25A37     9.260058e-07  9.999439
CREG1        9.260058e-07 10.025561
FAM214B      9.318818e-07 10.010975
TPGS2        1.004077e-06  9.926336
RILP         1.065672e-06  9.860041
MYL4         1.074095e-06  9.841866
ZRANB1       1.105781e-06  9.797233
MRC2         1.105781e-06  9.801204
PIP4K2A      1.136901e-06  9.756440
ASCC2        1.136901e-06  9.754852
ABCC13       1.145948e-06  9.549831
OPTN         1.151544e-06  9.715944
DMTN         1.151544e-06  9.704873
SELK         1.219698e-06  9.648031
CTD-2336O2.1 1.268519e-06  9.601385
SLC14A1      1.361679e-06  9.524409
HEMGN        1.411141e-06  9.469753</code></pre>
<pre class="r"><code>topTable(fit1, coef=2, adjust=&quot;BH&quot;, number=100, sort.by=&quot;T&quot;)</code></pre>
<pre><code>                      genes      logFC  AveExpr         t      P.Value
MPHOSPH8           MPHOSPH8  0.7168307 5.835914 10.904631 6.335906e-21
DFFA                   DFFA  0.5058557 4.197774  8.771917 2.909985e-15
RUNX3                 RUNX3  0.7284583 7.176955  8.436162 2.113642e-14
RTF1                   RTF1  0.5316677 5.220883  8.406322 2.517756e-14
ZNF791               ZNF791  0.5587761 3.870184  8.250105 6.269590e-14
UPF3A                 UPF3A  0.5498524 4.053190  8.158404 1.067990e-13
ANKRD12             ANKRD12  1.1236005 5.564936  7.955157 3.449790e-13
ZBTB7A               ZBTB7A  0.5133452 6.925620  7.886621 5.109487e-13
PURA                   PURA  0.6889979 2.421027  7.827782 7.150800e-13
RP11-83A24.2   RP11-83A24.2  0.9139026 2.609102  7.633004 2.159632e-12
WIPF2                 WIPF2  0.4082304 5.992864  7.606305 2.510633e-12
PRKAR2A             PRKAR2A  0.3888707 4.547113  7.497268 4.632988e-12
CCDC97               CCDC97  0.3498478 6.391842  7.413161 7.412382e-12
RSBN1L               RSBN1L  0.5140505 4.799770  7.375912 9.120507e-12
TTF1                   TTF1  0.4851441 3.157721  7.360008 9.963458e-12
POU2F1               POU2F1  0.6772434 4.742764  7.312371 1.297715e-11
GATA3                 GATA3  0.5440241 4.899425  7.262280 1.711963e-11
CCNK                   CCNK  0.3428886 6.120862  7.139244 3.368160e-11
RP11-473M20.7 RP11-473M20.7  0.5962701 6.654955  7.069163 4.940154e-11
TRIM38               TRIM38  0.3494019 5.922881  7.035986 5.918575e-11
AC009065.1       AC009065.1  0.6620303 3.270668  7.030425 6.100339e-11
CCAR1                 CCAR1  0.7416071 4.705804  7.027487 6.198561e-11
TRIM56               TRIM56  0.7885765 4.928674  6.912444 1.155754e-10
XIAP                   XIAP  0.4611892 4.321926  6.859613 1.535984e-10
RPL32P3             RPL32P3  0.4651906 3.564418  6.747793 2.794206e-10
AAK1                   AAK1  1.0107309 4.463210  6.744682 2.840902e-10
FAM133B             FAM133B  0.6182793 2.982537  6.657034 4.524640e-10
PRRC2C               PRRC2C  0.6440299 7.276964  6.650330 4.687997e-10
SH2D2A               SH2D2A  0.5671603 5.140647  6.530823 8.794863e-10
CWF19L2             CWF19L2  0.7106211 4.150955  6.524534 9.089396e-10
AC013394.2       AC013394.2 -0.5101620 4.760018 -6.493799 1.067461e-09
THOC2                 THOC2  0.6125518 4.859229  6.484216 1.122232e-09
RBBP6                 RBBP6  0.4267834 5.182964  6.450924 1.334903e-09
TRIM33               TRIM33  0.5293460 5.346465  6.420640 1.562516e-09
DEFA1                 DEFA1 -2.5170182 5.657122 -6.376151 1.967695e-09
NFIC                   NFIC  0.4439360 6.408937  6.297578 2.950399e-09
AGAP2                 AGAP2  0.3264907 6.446342  6.245226 3.858594e-09
DCP1A                 DCP1A  0.4256849 4.590691  6.223150 4.319322e-09
PHIP                   PHIP  0.6648982 4.880460  6.198257 4.903822e-09
PHACTR4             PHACTR4  0.4108943 4.389212  6.185575 5.230814e-09
AP000347.2       AP000347.2  0.7070365 3.398402  6.172899 5.579025e-09
MON1B                 MON1B  0.2670025 5.972489  6.146659 6.373755e-09
U2AF1                 U2AF1  0.2037132 6.157838  6.140502 6.575743e-09
NSRP1                 NSRP1  0.4333836 3.461913  6.130160 6.929309e-09
NCOR1                 NCOR1  0.4192418 6.518418  6.100237 8.060704e-09
RP11-357C3.3   RP11-357C3.3  0.4165877 4.988239  6.088927 8.534030e-09
ISG20L2             ISG20L2  0.2313524 5.121073  6.078919 8.975510e-09
RP11-582J16.5 RP11-582J16.5  0.4756860 2.871034  6.034060 1.124579e-08
DHX40                 DHX40 -0.3633279 3.906045 -6.004684 1.302856e-08
NACC1                 NACC1  0.3351195 5.828696  5.989764 1.403734e-08
COX17                 COX17 -0.5382919 2.819562 -5.943557 1.767255e-08
NUMBL                 NUMBL  0.3860862 3.077008  5.933171 1.860871e-08
APOBEC3D           APOBEC3D  0.3584945 4.768218  5.901891 2.173188e-08
EIF5B                 EIF5B  0.3733810 5.664326  5.889309 2.312808e-08
RPIA                   RPIA -0.7704579 4.349177 -5.884142 2.372663e-08
POU2F2               POU2F2  0.5751006 7.478788  5.871843 2.521304e-08
UNC50                 UNC50 -0.3487050 3.325949 -5.865225 2.605033e-08
TCOF1                 TCOF1  0.4090961 5.237552  5.825528 3.167454e-08
ZMYND8               ZMYND8  0.3359343 5.175443  5.793003 3.715490e-08
CCDC142             CCDC142  0.4729293 3.171367  5.782694 3.907839e-08
METTL9               METTL9 -0.4465140 6.407238 -5.766777 4.224089e-08
DLGAP4               DLGAP4  0.2535153 5.963165  5.732933 4.982081e-08
TRIP12               TRIP12 -0.3752266 7.022162 -5.728991 5.078569e-08
ANKRD11             ANKRD11  0.2663636 7.409060  5.725836 5.157121e-08
KIAA2026           KIAA2026  0.4658500 4.671033  5.717629 5.367086e-08
KIAA0430           KIAA0430 -0.4583816 7.202189 -5.713441 5.477422e-08
SAMD4B               SAMD4B  0.2865740 6.113255  5.707995 5.624245e-08
NACA                   NACA -0.3596938 8.422591 -5.690740 6.115256e-08
UPF3B                 UPF3B  0.5396277 2.978355  5.689387 6.155471e-08
BPI                     BPI -1.3983366 3.417495 -5.683815 6.323885e-08
CYP20A1             CYP20A1  0.4537031 2.093968  5.680157 6.436915e-08
APOBEC3F           APOBEC3F  0.3574236 3.417572  5.674964 6.600719e-08
GABARAPL2         GABARAPL2 -0.4702179 5.804466 -5.665856 6.897966e-08
DSTYK                 DSTYK  0.3159237 3.539352  5.656214 7.226900e-08
MBOAT2               MBOAT2 -0.6407015 4.339343 -5.639399 7.837579e-08
MT-ATP8             MT-ATP8  0.6710304 9.281045  5.638031 7.889417e-08
LTF                     LTF -1.8718550 3.944821 -5.637490 7.910006e-08
ACP1                   ACP1 -0.4734967 4.823921 -5.626020 8.359267e-08
VAMP2                 VAMP2  0.3455131 7.677016  5.611488 8.964355e-08
FECH                   FECH -1.3252207 5.170922 -5.607437 9.140550e-08
SEPT6                 SEPT6  0.3070434 7.936436  5.604099 9.288273e-08
CMIP                   CMIP  0.4391610 6.800712  5.595295 9.689113e-08
FAM46C               FAM46C -1.4109593 6.794919 -5.588661 1.000232e-07
SYNGAP1             SYNGAP1  0.4944706 5.591445  5.583648 1.024552e-07
ZC3H18               ZC3H18  0.3209911 6.067263  5.583390 1.025817e-07
UBAP2L               UBAP2L  0.2815273 6.882633  5.576020 1.062671e-07
DPF2                   DPF2  0.2884381 6.190302  5.568813 1.099958e-07
MBNL3                 MBNL3 -1.0082152 6.591863 -5.561004 1.141802e-07
DMWD                   DMWD  0.3792316 3.809142  5.558338 1.156438e-07
SESN3                 SESN3 -0.7551710 5.961494 -5.537746 1.275853e-07
IPO9                   IPO9  0.2510944 4.884354  5.531271 1.315827e-07
WBP4                   WBP4  0.4419867 2.866234  5.527188 1.341665e-07
UBQLN4               UBQLN4  0.2987111 5.042698  5.526528 1.345885e-07
TAF3                   TAF3  0.4496017 3.402842  5.506052 1.483529e-07
ITFG2                 ITFG2  0.2999735 4.844786  5.491799 1.587360e-07
SCLY                   SCLY  0.2748533 3.871607  5.477016 1.702539e-07
POLR2A               POLR2A  0.4244487 8.102956  5.476405 1.707465e-07
MEF2D                 MEF2D  0.2605754 6.409654  5.474064 1.726495e-07
TSPAN18             TSPAN18  0.3994628 4.859877  5.466801 1.786843e-07
PPARD                 PPARD  0.2883915 5.666312  5.448508 1.948130e-07
                 adj.P.Val         B
MPHOSPH8      7.247009e-17 36.609945
DFFA          1.664220e-11 24.032767
RUNX3         7.199524e-11 22.152542
RTF1          7.199524e-11 22.021006
ZNF791        1.434232e-10 21.062523
UPF3A         2.035944e-10 20.576691
ANKRD12       5.636958e-10 19.508863
ZBTB7A        7.305289e-10 19.116261
PURA          9.087872e-10 18.423952
RP11-83A24.2  2.470187e-09 17.449388
WIPF2         2.610602e-09 17.606976
PRKAR2A       4.416009e-09 17.004206
CCDC97        6.521756e-09 16.565641
RSBN1L        7.451454e-09 16.361944
TTF1          7.597468e-09 16.160412
POU2F1        9.277042e-09 16.024324
GATA3         1.151849e-08 15.762166
CCNK          2.140279e-08 15.117086
RP11-473M20.7 2.973973e-08 14.746567
TRIM38        3.222688e-08 14.577643
AC009065.1    3.222688e-08 14.482949
CCAR1         3.222688e-08 14.528376
TRIM56        5.747617e-08 13.934257
XIAP          7.320242e-08 13.654459
RPL32P3       1.249778e-07 13.056212
AAK1          1.249778e-07 13.071293
FAM133B       1.915047e-07 12.550101
PRRC2C        1.915047e-07 12.591646
SH2D2A        3.465484e-07 11.994771
CWF19L2       3.465484e-07 11.959284
AC013394.2    3.938585e-07 11.808845
THOC2         4.011279e-07 11.762012
RBBP6         4.626855e-07 11.596031
TRIM33        5.256488e-07 11.445428
DEFA1         6.430426e-07 11.226225
NFIC          9.374073e-07 10.839081
AGAP2         1.192827e-06 10.583003
DCP1A         1.300116e-06 10.477309
PHIP          1.438203e-06 10.355334
PHACTR4       1.495751e-06 10.295513
AP000347.2    1.556412e-06 10.222149
MON1B         1.735786e-06 10.102516
U2AF1         1.749148e-06 10.073192
NSRP1         1.801305e-06 10.019554
NCOR1         2.048852e-06  9.880974
RP11-357C3.3  2.122005e-06  9.826223
ISG20L2       2.184295e-06  9.777727
RP11-582J16.5 2.679779e-06  9.532039
DHX40         3.041237e-06  9.428632
NACC1         3.211182e-06  9.349255
COX17         3.963502e-06  9.078500
NUMBL         4.093200e-06  9.074126
APOBEC3D      4.689986e-06  8.938193
EIF5B         4.898869e-06  8.873320
RPIA          4.934276e-06  8.861239
POU2F2        5.149764e-06  8.802612
UNC50         5.227432e-06  8.763348
TCOF1         6.246438e-06  8.575281
ZMYND8        7.203012e-06  8.424022
CCDC142       7.449644e-06  8.382069
METTL9        7.920513e-06  8.301148
DLGAP4        9.191135e-06  8.142788
TRIP12        9.216742e-06  8.132631
ANKRD11       9.216742e-06  8.122993
KIAA2026      9.444419e-06  8.080136
KIAA0430      9.492538e-06  8.062584
SAMD4B        9.601509e-06  8.028210
NACA          1.020381e-05  7.967616
UPF3B         1.020381e-05  7.953419
BPI           1.033323e-05  7.930152
CYP20A1       1.036978e-05  7.810425
APOBEC3F      1.048598e-05  7.896204
GABARAPL2     1.080807e-05  7.832831
DSTYK         1.117044e-05  7.811372
MBOAT2        1.174996e-05  7.730333
MT-ATP8       1.174996e-05  7.728131
LTF           1.174996e-05  7.726363
ACP1          1.225811e-05  7.660251
VAMP2         1.297902e-05  7.602170
FECH          1.306870e-05  7.572515
SEPT6         1.311596e-05  7.571296
CMIP          1.351513e-05  7.518621
FAM46C        1.378392e-05  7.485709
SYNGAP1       1.380388e-05  7.457029
ZC3H18        1.380388e-05  7.456779
UBAP2L        1.413352e-05  7.431810
DPF2          1.446128e-05  7.390937
MBNL3         1.484083e-05  7.357666
DMWD          1.486218e-05  7.367200
SESN3         1.621468e-05  7.248255
IPO9          1.653893e-05  7.225798
WBP4          1.655294e-05  7.218804
UBQLN4        1.655294e-05  7.201698
TAF3          1.805171e-05  7.135467
ITFG2         1.911181e-05  7.048204
SCLY          2.013401e-05  7.001205
POLR2A        2.013401e-05  6.998230
MEF2D         2.015067e-05  6.964966
TSPAN18       2.064435e-05  6.935484
PPARD         2.228271e-05  6.846172</code></pre>
</div>
<div id="topgo-enrichment" class="section level1">
<h1>TopGO enrichment</h1>
<pre class="r"><code>#time_12 &lt;- Time1v2$adj.P.Val
#names(time_12) &lt;- rownames(Time1v2)
#time_23 &lt;- Time2v3$adj.P.Val
#names(time_23) &lt;- rownames(Time2v3)

# Run topGO on time 1 versus 2
#go_data &lt;- new(&quot;topGOdata&quot;,
#                   ontology = &quot;BP&quot;,
#                   allGenes = time_12, 
#                    geneSel = function(allScore){
#    return(allScore &lt; 0.01)
#},
#                   nodeSize = 5,
#                   annotationFun = annFUN.org,
#                   mapping = &quot;org.Hs.eg.db&quot;,
#                   ID = &quot;ensembl&quot;)

# Perform enrichment test
#go_test &lt;- runTest(go_data, algorithm = &quot;weight01&quot;, statistic = &quot;fisher&quot;)

#go_table &lt;- GenTable(go_data, weightFisher = go_test,
#                         orderBy = &quot;weightFisher&quot;, ranksOf = &quot;weightFisher&quot;,
#                         topNodes = sum(score(go_test) &lt; .01))

#go_table




# Run topGO on time 2 versus 3
#go_data &lt;- new(&quot;topGOdata&quot;,
#                   ontology = &quot;BP&quot;,
#                   allGenes = time_12, 
#                    geneSel = function(allScore){
#    return(allScore &lt; 0.01)
#},
#                   nodeSize = 5,
#                   annotationFun = annFUN.org,
#                   mapping = &quot;org.Hs.eg.db&quot;,
#                   ID = &quot;ensembl&quot;)

# Perform enrichment test
#go_test &lt;- runTest(go_data, algorithm = &quot;weight01&quot;, statistic = &quot;fisher&quot;)

#go_table &lt;- GenTable(go_data, weightFisher = go_test,
#                         orderBy = &quot;weightFisher&quot;, ranksOf = &quot;weightFisher&quot;,
#                         topNodes = sum(score(go_test) &lt; .01))

#go_table</code></pre>
</div>
<div id="differential-expression-analysis-with-deseq2" class="section level1">
<h1>Differential expression analysis with DESeq2</h1>
<pre class="r"><code>library(&quot;vsn&quot;)
library(&quot;tidyverse&quot;)
library(&quot;RColorBrewer&quot;)
library(&quot;pheatmap&quot;)</code></pre>
<pre><code>Warning: package &#39;pheatmap&#39; was built under R version 3.4.4</code></pre>
<pre class="r"><code>library(&quot;BiocParallel&quot;)
register(MulticoreParam(4))

## Set color palette for figures
colors &lt;- colorRampPalette( rev(brewer.pal(9, &quot;Blues&quot;)) )(255)

tx.salmon &lt;- tximport(salmon.files, type = &quot;salmon&quot;, tx2gene = tx2gene, dropInfReps=TRUE)</code></pre>
<pre><code>reading in files with read_tsv</code></pre>
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 
summarizing abundance
summarizing counts
summarizing length</code></pre>
<pre class="r"><code>dds &lt;- DESeqDataSetFromTximport(tx.salmon, colData=clinical_sample, design = ~ BAN_ID + Time)</code></pre>
<pre><code>using counts and average transcript lengths from tximport</code></pre>
<pre class="r"><code># Minimal filtering

#dds &lt;- dds[ rowSums(counts(dds)) &gt; 76, ]
#nrow(dds)

# Visualize the data

ntd &lt;- normTransform(dds)</code></pre>
<pre><code>using &#39;avgTxLength&#39; from assays(dds), correcting for library size</code></pre>
<pre class="r"><code>meanSdPlot(assay(ntd))</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>vsd &lt;- vst(dds, blind = FALSE)</code></pre>
<pre><code>using &#39;avgTxLength&#39; from assays(dds), correcting for library size</code></pre>
<pre class="r"><code>head(assay(vsd),3)</code></pre>
<pre><code>        BAN22201T1 BAN22201T2 BAN22201T3 BAN22202T1 BAN22202T2 BAN22202T3
5S_rRNA   5.214418   5.214418   5.476612   5.214418   5.463526   5.214418
7SK       7.551367   7.975540   8.460954   8.371897   7.651294   7.426007
A1BG      7.174287   7.133234   7.207594   7.329056   6.971962   7.103051
        BAN22203T1 BAN22203T2 BAN22204T1 BAN22204T2 BAN22204T3 BAN22205T1
5S_rRNA   5.214418   5.214418   5.214418   5.731319   5.214418   5.396193
7SK       8.461531   7.547555   7.651979   8.376073   8.138787   8.009698
A1BG      6.952058   6.687685   6.908285   7.240791   6.755725   7.436359
        BAN22205T2 BAN22206T1 BAN22206T2 BAN22206T3 BAN22207T1 BAN22207T2
5S_rRNA   5.214418   5.214418   5.214418   5.484722   5.214418   5.614141
7SK       7.425812   8.014030   8.140603   6.454187   7.086543   8.030510
A1BG      6.771926   7.136523   6.803289   6.619033   7.395910   6.889429
        BAN22207T3 BAN22208T1 BAN22208T2 BAN22208T3 BAN22209T1 BAN22209T2
5S_rRNA   5.530835   5.214418   5.516591   5.214418   5.214418   5.499223
7SK       6.713135   8.161133   9.115627   7.820813   8.029746   7.830516
A1BG      6.940259   6.736633   6.945530   7.110813   6.979532   7.405830
        BAN22209T3 BAN22210T1 BAN22210T2 BAN22210T3 BAN22212T1 BAN22212T2
5S_rRNA   5.214418   5.214418   5.214418   5.652451   5.584088   5.535227
7SK       7.680964   7.977000   8.328045   8.230283   8.019961   8.477948
A1BG      7.226887   7.361029   6.912841   6.842325   7.223889   7.119062
        BAN22215T1 BAN22215T2 BAN22215T3 BAN22216T1 BAN22216T2 BAN22216T3
5S_rRNA   5.214418   5.602927   5.214418   5.435022   5.214418   5.548038
7SK       8.099599   7.498618   8.651010   8.123251   7.847659   8.297309
A1BG      6.811245   6.938667   7.128091   6.864543   6.672707   7.126519
        BAN22218T1 BAN22218T2 BAN22218T3 BAN22219T1 BAN22219T2 BAN22219T3
5S_rRNA   5.214418   5.461453   6.021834   5.399143   5.214418   5.214418
7SK       8.318324   7.725166   8.175284   7.427576   7.820354   7.695150
A1BG      7.057952   6.946430   7.106593   6.874990   6.771940   6.907873
        BAN22220T1 BAN22220T2 BAN22220T3 BAN22221T1 BAN22221T2 BAN22221T3
5S_rRNA   5.214418   5.214418   5.214418   5.214418   5.505725   5.214418
7SK       8.130972   9.107222   8.473389   7.819076   7.843332   8.136712
A1BG      7.166290   7.651324   7.374661   7.248954   7.435505   7.290777
        BAN22222T1 BAN22222T2 BAN22224T1 BAN22224T2 BAN22224T3 BAN22226T1
5S_rRNA   5.251663   5.672349   5.214418   5.938330   5.214418   5.519110
7SK       7.486916   8.089132   8.929906   9.144003   7.980444   7.960335
A1BG      8.413605   7.052438   8.277304   7.108239   6.975739   7.420103
        BAN22226T2 BAN22226T3 BAN22228T1 BAN22228T2 BAN22228T3 BAN22229T1
5S_rRNA   5.444931   5.214418   5.633511   5.868410   5.575372   5.468007
7SK       8.097299   8.265523   8.178422   7.548322   7.957784   6.889453
A1BG      7.461807   6.851058   7.277687   7.419139   6.763190   7.673365
        BAN22229T2 BAN22232T1 BAN22232T2 BAN22232T3 BAN22233T1 BAN22233T2
5S_rRNA   5.214418   5.825625   6.350020   5.380430   5.214418   5.655171
7SK       8.530010   6.386326   8.767203   8.240192   7.584256   7.883854
A1BG      7.316627   7.747207   7.328394   6.622784   7.175602   6.902189
        BAN22233T3 BAN22234T1 BAN22234T2 BAN22234T3 BAN22235T1 BAN22235T2
5S_rRNA   5.214418   5.214418   5.214418   5.214418   5.214418   5.214418
7SK       7.746251   7.915018   8.403332   8.632593   7.486339   7.927732
A1BG      7.156510   7.014711   6.862827   6.825939   7.034322   6.486998
        BAN22235T3 BAN22236T1 BAN22236T2 BAN22239T1 BAN22239T2 BAN22239T3
5S_rRNA   5.426660   5.214418   5.214418   5.418858   5.611557   5.214418
7SK       8.203257   7.454961   6.529618   7.506168   8.770131   8.202734
A1BG      7.223341   7.000550   7.057217   6.966197   7.341928   7.295739
        BAN22240T1 BAN22240T2 BAN22242T1 BAN22242T2 BAN22242T3 BAN22243T1
5S_rRNA   5.214418   5.548321   5.214418   5.214418   5.214418   5.214418
7SK       7.744576   7.933310   7.808706   8.317617   8.340620   8.598055
A1BG      6.882977   7.524839   7.725187   7.222289   7.936624   7.037910
        BAN22243T2 BAN22243T3 BAN22245T1 BAN22245T2 BAN22247T1 BAN22247T2
5S_rRNA   5.601765   5.214418   5.483847   5.539412   5.214418   5.646931
7SK       8.154298   8.844134   8.315088   8.559215   8.223336   8.323649
A1BG      6.899104   7.242056   6.953467   7.149965   7.461467   7.384766
        BAN22248T1 BAN22248T2 BAN22248T3 BAN22249T1 BAN22249T2 BAN22250T1
5S_rRNA   5.479652   5.480198   5.798497   5.452252   5.701196   5.736343
7SK       8.319636   8.342455   8.777110   7.958410   8.125030   8.042225
A1BG      6.942216   6.840899   6.956959   7.408890   7.878244   7.561740
        BAN22250T2 BAN22251T1 BAN22251T2 BAN22252T1 BAN22252T2 BAN22253T1
5S_rRNA   5.540136   5.918574   5.214418   5.939406   5.506647   5.472196
7SK       8.594365   8.463487   7.724802   7.103463   9.022052   8.409917
A1BG      7.625660   7.452029   7.516614   7.519051   7.110940   7.537311
        BAN22253T2 BAN22254T1 BAN22254T2 BAN22254T3 BAN22255T1 BAN22255T2
5S_rRNA   5.214418   5.641606   5.443517   5.214418   5.408205   5.214418
7SK       9.051799   8.323733   8.323914   7.756964   7.910018   8.028087
A1BG      6.786041   8.280038   7.104394   6.610562   6.958128   6.939246
        BAN22256T1 BAN22256T2 BAN22256T3 BAN22257T1 BAN22257T2 BAN22258T1
5S_rRNA   5.214418   5.214418   5.214418   5.214418   5.214418   5.214418
7SK       7.978313   7.954968   8.197652   8.361582   7.756202   8.368986
A1BG      7.218380   7.481178   7.626978   7.473319   7.235610   6.963338
        BAN22258T2 BAN22258T3 BAN22260T1 BAN22260T2 BAN22260T3 BAN22261T1
5S_rRNA   5.214418   5.508942   5.518838   5.498032   5.434250   5.214418
7SK       8.124486   6.268826   8.395860   9.198374   7.754274   7.995214
A1BG      6.999856   7.988881   6.984447   7.302465   7.267606   7.373140
        BAN22261T2 BAN22262T1 BAN22262T2 BAN22266T1 BAN22266T2 BAN22266T3
5S_rRNA   5.214418   5.520244   5.214418   5.214418   5.584931   5.214418
7SK       8.051803   7.761467   8.744429   8.452440   7.967749   9.008938
A1BG      7.421706   7.219127   7.140857   7.192493   8.265142   6.898346
        BAN22267T1 BAN22267T2 BAN22268T1 BAN22268T2 BAN22268T3 BAN22269T1
5S_rRNA   5.214418   5.214418   5.214418   5.641242   5.214418   5.214418
7SK       8.548275   8.140419   8.584197   8.291240   7.079052   8.215200
A1BG      6.920829   7.706743   7.440870   7.343239   6.568856   7.450902
        BAN22269T2 BAN22270T1 BAN22270T2 BAN22271T1 BAN22271T2 BAN22271T3
5S_rRNA   5.214418   5.214418   5.214418   5.214418   5.435174   5.214418
7SK       7.830820   7.824386   7.536540   8.501970   8.982355   8.660374
A1BG      8.071468   7.166642   7.528644   7.429863   7.463978   6.782824
        BAN22272T1 BAN22272T2 BAN22272T3 BAN22274T1 BAN22274T2 BAN22274T3
5S_rRNA   5.214418   5.214418   5.214418   5.214418   5.420498   5.396523
7SK       8.510441   8.121057   7.420216   7.987898   8.167034   8.179500
A1BG      7.046400   6.840580   7.188196   7.084175   7.376835   5.688983
        BAN22202_RRED397T1 BAN22202_RRED397T2 BAN22220_RRED398T1
5S_rRNA           5.214418           5.214418           5.214418
7SK               8.182834           8.566705           8.842481
A1BG              8.084313           8.101655           8.357417
        BAN22220_RRED398T2 BAN22209_RRED399T1 BAN22209_RRED399T2
5S_rRNA           5.214418           5.214418           5.214418
7SK               8.509809           8.631353           8.356687
A1BG              8.214921           8.579835           8.675117
        BAN22218_RRED424T1 BAN22218_RRED424T2 BAN22226_RRED453T1
5S_rRNA           5.214418           5.214418           5.558093
7SK               8.381989           8.678157           8.721914
A1BG              8.626399           9.421711           8.205643
        BAN22226_RRED453T2 BAN22228_RRED467T1 BAN22228_RRED467T2
5S_rRNA           5.214418           5.214418           5.214418
7SK               8.391908           8.235824           8.605649
A1BG              8.893250           8.071887           8.375545</code></pre>
<pre class="r"><code>vsd_true &lt;- vst(dds, blind = TRUE)</code></pre>
<pre><code>using &#39;avgTxLength&#39; from assays(dds), correcting for library size</code></pre>
<pre class="r"><code>head(assay(vsd),3)</code></pre>
<pre><code>        BAN22201T1 BAN22201T2 BAN22201T3 BAN22202T1 BAN22202T2 BAN22202T3
5S_rRNA   5.214418   5.214418   5.476612   5.214418   5.463526   5.214418
7SK       7.551367   7.975540   8.460954   8.371897   7.651294   7.426007
A1BG      7.174287   7.133234   7.207594   7.329056   6.971962   7.103051
        BAN22203T1 BAN22203T2 BAN22204T1 BAN22204T2 BAN22204T3 BAN22205T1
5S_rRNA   5.214418   5.214418   5.214418   5.731319   5.214418   5.396193
7SK       8.461531   7.547555   7.651979   8.376073   8.138787   8.009698
A1BG      6.952058   6.687685   6.908285   7.240791   6.755725   7.436359
        BAN22205T2 BAN22206T1 BAN22206T2 BAN22206T3 BAN22207T1 BAN22207T2
5S_rRNA   5.214418   5.214418   5.214418   5.484722   5.214418   5.614141
7SK       7.425812   8.014030   8.140603   6.454187   7.086543   8.030510
A1BG      6.771926   7.136523   6.803289   6.619033   7.395910   6.889429
        BAN22207T3 BAN22208T1 BAN22208T2 BAN22208T3 BAN22209T1 BAN22209T2
5S_rRNA   5.530835   5.214418   5.516591   5.214418   5.214418   5.499223
7SK       6.713135   8.161133   9.115627   7.820813   8.029746   7.830516
A1BG      6.940259   6.736633   6.945530   7.110813   6.979532   7.405830
        BAN22209T3 BAN22210T1 BAN22210T2 BAN22210T3 BAN22212T1 BAN22212T2
5S_rRNA   5.214418   5.214418   5.214418   5.652451   5.584088   5.535227
7SK       7.680964   7.977000   8.328045   8.230283   8.019961   8.477948
A1BG      7.226887   7.361029   6.912841   6.842325   7.223889   7.119062
        BAN22215T1 BAN22215T2 BAN22215T3 BAN22216T1 BAN22216T2 BAN22216T3
5S_rRNA   5.214418   5.602927   5.214418   5.435022   5.214418   5.548038
7SK       8.099599   7.498618   8.651010   8.123251   7.847659   8.297309
A1BG      6.811245   6.938667   7.128091   6.864543   6.672707   7.126519
        BAN22218T1 BAN22218T2 BAN22218T3 BAN22219T1 BAN22219T2 BAN22219T3
5S_rRNA   5.214418   5.461453   6.021834   5.399143   5.214418   5.214418
7SK       8.318324   7.725166   8.175284   7.427576   7.820354   7.695150
A1BG      7.057952   6.946430   7.106593   6.874990   6.771940   6.907873
        BAN22220T1 BAN22220T2 BAN22220T3 BAN22221T1 BAN22221T2 BAN22221T3
5S_rRNA   5.214418   5.214418   5.214418   5.214418   5.505725   5.214418
7SK       8.130972   9.107222   8.473389   7.819076   7.843332   8.136712
A1BG      7.166290   7.651324   7.374661   7.248954   7.435505   7.290777
        BAN22222T1 BAN22222T2 BAN22224T1 BAN22224T2 BAN22224T3 BAN22226T1
5S_rRNA   5.251663   5.672349   5.214418   5.938330   5.214418   5.519110
7SK       7.486916   8.089132   8.929906   9.144003   7.980444   7.960335
A1BG      8.413605   7.052438   8.277304   7.108239   6.975739   7.420103
        BAN22226T2 BAN22226T3 BAN22228T1 BAN22228T2 BAN22228T3 BAN22229T1
5S_rRNA   5.444931   5.214418   5.633511   5.868410   5.575372   5.468007
7SK       8.097299   8.265523   8.178422   7.548322   7.957784   6.889453
A1BG      7.461807   6.851058   7.277687   7.419139   6.763190   7.673365
        BAN22229T2 BAN22232T1 BAN22232T2 BAN22232T3 BAN22233T1 BAN22233T2
5S_rRNA   5.214418   5.825625   6.350020   5.380430   5.214418   5.655171
7SK       8.530010   6.386326   8.767203   8.240192   7.584256   7.883854
A1BG      7.316627   7.747207   7.328394   6.622784   7.175602   6.902189
        BAN22233T3 BAN22234T1 BAN22234T2 BAN22234T3 BAN22235T1 BAN22235T2
5S_rRNA   5.214418   5.214418   5.214418   5.214418   5.214418   5.214418
7SK       7.746251   7.915018   8.403332   8.632593   7.486339   7.927732
A1BG      7.156510   7.014711   6.862827   6.825939   7.034322   6.486998
        BAN22235T3 BAN22236T1 BAN22236T2 BAN22239T1 BAN22239T2 BAN22239T3
5S_rRNA   5.426660   5.214418   5.214418   5.418858   5.611557   5.214418
7SK       8.203257   7.454961   6.529618   7.506168   8.770131   8.202734
A1BG      7.223341   7.000550   7.057217   6.966197   7.341928   7.295739
        BAN22240T1 BAN22240T2 BAN22242T1 BAN22242T2 BAN22242T3 BAN22243T1
5S_rRNA   5.214418   5.548321   5.214418   5.214418   5.214418   5.214418
7SK       7.744576   7.933310   7.808706   8.317617   8.340620   8.598055
A1BG      6.882977   7.524839   7.725187   7.222289   7.936624   7.037910
        BAN22243T2 BAN22243T3 BAN22245T1 BAN22245T2 BAN22247T1 BAN22247T2
5S_rRNA   5.601765   5.214418   5.483847   5.539412   5.214418   5.646931
7SK       8.154298   8.844134   8.315088   8.559215   8.223336   8.323649
A1BG      6.899104   7.242056   6.953467   7.149965   7.461467   7.384766
        BAN22248T1 BAN22248T2 BAN22248T3 BAN22249T1 BAN22249T2 BAN22250T1
5S_rRNA   5.479652   5.480198   5.798497   5.452252   5.701196   5.736343
7SK       8.319636   8.342455   8.777110   7.958410   8.125030   8.042225
A1BG      6.942216   6.840899   6.956959   7.408890   7.878244   7.561740
        BAN22250T2 BAN22251T1 BAN22251T2 BAN22252T1 BAN22252T2 BAN22253T1
5S_rRNA   5.540136   5.918574   5.214418   5.939406   5.506647   5.472196
7SK       8.594365   8.463487   7.724802   7.103463   9.022052   8.409917
A1BG      7.625660   7.452029   7.516614   7.519051   7.110940   7.537311
        BAN22253T2 BAN22254T1 BAN22254T2 BAN22254T3 BAN22255T1 BAN22255T2
5S_rRNA   5.214418   5.641606   5.443517   5.214418   5.408205   5.214418
7SK       9.051799   8.323733   8.323914   7.756964   7.910018   8.028087
A1BG      6.786041   8.280038   7.104394   6.610562   6.958128   6.939246
        BAN22256T1 BAN22256T2 BAN22256T3 BAN22257T1 BAN22257T2 BAN22258T1
5S_rRNA   5.214418   5.214418   5.214418   5.214418   5.214418   5.214418
7SK       7.978313   7.954968   8.197652   8.361582   7.756202   8.368986
A1BG      7.218380   7.481178   7.626978   7.473319   7.235610   6.963338
        BAN22258T2 BAN22258T3 BAN22260T1 BAN22260T2 BAN22260T3 BAN22261T1
5S_rRNA   5.214418   5.508942   5.518838   5.498032   5.434250   5.214418
7SK       8.124486   6.268826   8.395860   9.198374   7.754274   7.995214
A1BG      6.999856   7.988881   6.984447   7.302465   7.267606   7.373140
        BAN22261T2 BAN22262T1 BAN22262T2 BAN22266T1 BAN22266T2 BAN22266T3
5S_rRNA   5.214418   5.520244   5.214418   5.214418   5.584931   5.214418
7SK       8.051803   7.761467   8.744429   8.452440   7.967749   9.008938
A1BG      7.421706   7.219127   7.140857   7.192493   8.265142   6.898346
        BAN22267T1 BAN22267T2 BAN22268T1 BAN22268T2 BAN22268T3 BAN22269T1
5S_rRNA   5.214418   5.214418   5.214418   5.641242   5.214418   5.214418
7SK       8.548275   8.140419   8.584197   8.291240   7.079052   8.215200
A1BG      6.920829   7.706743   7.440870   7.343239   6.568856   7.450902
        BAN22269T2 BAN22270T1 BAN22270T2 BAN22271T1 BAN22271T2 BAN22271T3
5S_rRNA   5.214418   5.214418   5.214418   5.214418   5.435174   5.214418
7SK       7.830820   7.824386   7.536540   8.501970   8.982355   8.660374
A1BG      8.071468   7.166642   7.528644   7.429863   7.463978   6.782824
        BAN22272T1 BAN22272T2 BAN22272T3 BAN22274T1 BAN22274T2 BAN22274T3
5S_rRNA   5.214418   5.214418   5.214418   5.214418   5.420498   5.396523
7SK       8.510441   8.121057   7.420216   7.987898   8.167034   8.179500
A1BG      7.046400   6.840580   7.188196   7.084175   7.376835   5.688983
        BAN22202_RRED397T1 BAN22202_RRED397T2 BAN22220_RRED398T1
5S_rRNA           5.214418           5.214418           5.214418
7SK               8.182834           8.566705           8.842481
A1BG              8.084313           8.101655           8.357417
        BAN22220_RRED398T2 BAN22209_RRED399T1 BAN22209_RRED399T2
5S_rRNA           5.214418           5.214418           5.214418
7SK               8.509809           8.631353           8.356687
A1BG              8.214921           8.579835           8.675117
        BAN22218_RRED424T1 BAN22218_RRED424T2 BAN22226_RRED453T1
5S_rRNA           5.214418           5.214418           5.558093
7SK               8.381989           8.678157           8.721914
A1BG              8.626399           9.421711           8.205643
        BAN22226_RRED453T2 BAN22228_RRED467T1 BAN22228_RRED467T2
5S_rRNA           5.214418           5.214418           5.214418
7SK               8.391908           8.235824           8.605649
A1BG              8.893250           8.071887           8.375545</code></pre>
<pre class="r"><code>meanSdPlot(assay(vsd))</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># As stated in the DESeq2 manual, rlog takes a long time
#rld &lt;- rlog(dds, blind = FALSE)
#meanSdPlot(assay(rld))

# Distance based on roughly equal contribution from each gene

sampleDists &lt;- dist(t(assay(vsd)))
head(sampleDists)</code></pre>
<pre><code>[1] 44.61745 69.55544 66.63311 61.73040 59.12100 69.16001</code></pre>
<pre class="r"><code>sampleDistMatrix &lt;- as.matrix( sampleDists )
rownames(sampleDistMatrix) &lt;- paste(vsd$Time)
colnames(sampleDistMatrix) &lt;- NULL
colors &lt;- colorRampPalette( rev(brewer.pal(9, &quot;Blues&quot;)) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotPCA(vsd, intgroup=c(&quot;Time&quot;))</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-11-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Looking at outliers

#boxplot(log10(assays(dds)[[&quot;cooks&quot;]]), range = 0, las=2)

# Dispersion plot

#plotDispEsts(dds)


pcaData &lt;- plotPCA(vsd, intgroup=c(&quot;Time&quot;), returnData = TRUE)
percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))

ggplot(pcaData, aes(x = PC1, y = PC2, color = Time)) +
  geom_point(size =3) +
  xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) +
  coord_fixed()</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-11-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pcaData &lt;- plotPCA(vsd_true, intgroup=c(&quot;Time&quot;), returnData = TRUE)
percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))

ggplot(pcaData, aes(x = PC1, y = PC2, color = Time)) +
  geom_point(size =3) +
  xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) +
  coord_fixed()</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-11-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>X = t(scale(t(assay(vsd)),center=TRUE,scale=FALSE))
sv = svd(t(X))
U = sv$u
V = sv$v
D = sv$d
## in R calculate the rank of a matrix is by
#qr(t(X))$rank

plot(U[,1],U[,2],xlab=&quot;PC1&quot;,ylab=&quot;PC2&quot;)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-11-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>U &lt;- as.data.frame(U)
ggplot(data=U, aes(x=U[,1], y=U[,2], color=clinical_sample[,2])) + geom_point(aes(colour = as.factor(clinical_sample[,2]))) + ggtitle(&quot;PCA of normalized counts&quot;) + scale_color_discrete(name = &quot;Time&quot;) + xlab(&quot;PC1&quot;) + ylab(&quot;PC2&quot;)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-11-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>X = t(scale(t(assay(vsd_true)),center=TRUE,scale=FALSE))
sv = svd(t(X))
U = sv$u
V = sv$v
D = sv$d
## in R calculate the rank of a matrix is by
#qr(t(X))$rank

plot(U[,1],U[,2],xlab=&quot;PC1&quot;,ylab=&quot;PC2&quot;)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-11-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>U &lt;- as.data.frame(U)
ggplot(data=U, aes(x=U[,1], y=U[,2], color=clinical_sample[,2])) + geom_point(aes(colour = as.factor(clinical_sample[,2]))) + ggtitle(&quot;PCA of normalized counts&quot;) + scale_color_discrete(name = &quot;Time&quot;) + xlab(&quot;PC1&quot;) + ylab(&quot;PC2&quot;)</code></pre>
<p><img src="figure/import_transcript_level_estimates.Rmd/unnamed-chunk-11-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Write normalized counts to file
#normalized.counts &lt;- as.data.frame(counts(dds, normalized=TRUE ))
#head(normalized.counts)</code></pre>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.3 (2017-11-30)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: OS X El Capitan 10.11.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] hexbin_1.27.2              BiocParallel_1.12.0       
 [3] pheatmap_1.0.10            RColorBrewer_1.1-2        
 [5] vsn_3.46.0                 edgeR_3.20.9              
 [7] limma_3.34.9               EnsDb.Hsapiens.v75_2.99.0 
 [9] ensembldb_2.2.2            AnnotationFilter_1.2.0    
[11] GenomicFeatures_1.30.3     plyr_1.8.4                
[13] data.table_1.11.4          forcats_0.3.0             
[15] stringr_1.3.1              dplyr_0.7.6               
[17] purrr_0.2.5                tidyr_0.8.1               
[19] tibble_1.4.2               tidyverse_1.2.1           
[21] org.Hs.eg.db_3.5.0         clusterProfiler_3.6.0     
[23] DOSE_3.4.0                 topGO_2.30.1              
[25] SparseM_1.77               GO.db_3.5.0               
[27] AnnotationDbi_1.40.0       graph_1.56.0              
[29] cowplot_0.9.3              ggplot2_3.0.0             
[31] DESeq2_1.18.1              SummarizedExperiment_1.8.1
[33] DelayedArray_0.4.1         matrixStats_0.54.0        
[35] Biobase_2.38.0             GenomicRanges_1.30.3      
[37] GenomeInfoDb_1.14.0        IRanges_2.12.0            
[39] S4Vectors_0.16.0           BiocGenerics_0.24.0       
[41] readr_1.1.1                tximport_1.6.0            

loaded via a namespace (and not attached):
  [1] readxl_1.1.0                  backports_1.1.2              
  [3] Hmisc_4.1-1                   AnnotationHub_2.10.1         
  [5] fastmatch_1.1-0               workflowr_1.1.1              
  [7] igraph_1.2.2                  lazyeval_0.2.1               
  [9] splines_3.4.3                 digest_0.6.15                
 [11] BiocInstaller_1.28.0          htmltools_0.3.6              
 [13] GOSemSim_2.4.1                magrittr_1.5                 
 [15] checkmate_1.8.5               memoise_1.1.0                
 [17] cluster_2.0.7-1               Biostrings_2.46.0            
 [19] annotate_1.56.2               modelr_0.1.2                 
 [21] R.utils_2.6.0                 prettyunits_1.0.2            
 [23] colorspace_1.3-2              blob_1.1.1                   
 [25] rvest_0.3.2                   haven_1.1.2                  
 [27] crayon_1.3.4                  RCurl_1.95-4.11              
 [29] jsonlite_1.5                  genefilter_1.60.0            
 [31] bindr_0.1.1                   survival_2.42-6              
 [33] glue_1.3.0                    gtable_0.2.0                 
 [35] zlibbioc_1.24.0               XVector_0.18.0               
 [37] scales_0.5.0                  DBI_1.0.0                    
 [39] Rcpp_0.12.18                  xtable_1.8-2                 
 [41] progress_1.2.0                htmlTable_1.12               
 [43] foreign_0.8-71                bit_1.1-14                   
 [45] preprocessCore_1.40.0         Formula_1.2-3                
 [47] htmlwidgets_1.2               httr_1.3.1                   
 [49] fgsea_1.4.1                   acepack_1.4.1                
 [51] pkgconfig_2.0.1               XML_3.98-1.13                
 [53] R.methodsS3_1.7.1             nnet_7.3-12                  
 [55] locfit_1.5-9.1                labeling_0.3                 
 [57] tidyselect_0.2.4              rlang_0.2.1                  
 [59] reshape2_1.4.3                later_0.7.3                  
 [61] munsell_0.5.0                 cellranger_1.1.0             
 [63] tools_3.4.3                   cli_1.0.0                    
 [65] RSQLite_2.1.1                 broom_0.5.0                  
 [67] evaluate_0.11                 yaml_2.2.0                   
 [69] knitr_1.20                    bit64_0.9-7                  
 [71] bindrcpp_0.2.2                nlme_3.1-137                 
 [73] mime_0.5                      whisker_0.3-2                
 [75] R.oo_1.22.0                   DO.db_2.9                    
 [77] xml2_1.2.0                    biomaRt_2.34.2               
 [79] compiler_3.4.3                rstudioapi_0.7               
 [81] curl_3.2                      interactiveDisplayBase_1.16.0
 [83] affyio_1.48.0                 geneplotter_1.56.0           
 [85] stringi_1.2.4                 lattice_0.20-35              
 [87] ProtGenerics_1.10.0           Matrix_1.2-14                
 [89] pillar_1.3.0                  bitops_1.0-6                 
 [91] httpuv_1.4.5                  rtracklayer_1.38.3           
 [93] qvalue_2.10.0                 affy_1.56.0                  
 [95] R6_2.2.2                      latticeExtra_0.6-28          
 [97] promises_1.0.1                RMySQL_0.10.15               
 [99] gridExtra_2.3                 assertthat_0.2.0             
[101] rprojroot_1.3-2               withr_2.1.2                  
[103] GenomicAlignments_1.14.2      Rsamtools_1.30.0             
[105] GenomeInfoDbData_1.0.0        hms_0.4.2                    
[107] grid_3.4.3                    rpart_4.1-13                 
[109] rmarkdown_1.10                rvcheck_0.1.0                
[111] git2r_0.23.0                  shiny_1.1.0                  
[113] lubridate_1.7.4               base64enc_0.1-3              </code></pre>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
